<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1 Árboles filogenéticos. Borrador | El manual de R para la Epidemiología</title>
<meta name="author" content="El equipo del manual">
<meta name="description" content="1.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="1 Árboles filogenéticos. Borrador | El manual de R para la Epidemiología">
<meta property="og:type" content="book">
<meta property="og:description" content="1.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="1 Árboles filogenéticos. Borrador | El manual de R para la Epidemiología">
<meta name="twitter:description" content="1.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.19/datatables.js"></script><link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="style_bs4.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">El manual de R para la Epidemiología</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"></a></li>
<li class="book-part">Preview pages</li>
<li><a class="active" href="phylogenetic-trees-1.html"><span class="header-section-number">1</span> Árboles filogenéticos. Borrador</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="phylogenetic-trees-1" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Árboles filogenéticos. Borrador<a class="anchor" aria-label="anchor" href="#phylogenetic-trees-1"><i class="fas fa-link"></i></a>
</h1>
<!-- ======================================================= -->
<div id="overview-7" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Resumen<a class="anchor" aria-label="anchor" href="#overview-7"><i class="fas fa-link"></i></a>
</h2>
<p>Los <strong>árboles filogenéticos</strong> se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético.</p>
<p>Pueden construirse a partir de secuencias genéticas utilizando métodos basados en la distancia (como el método de unión de vecinos) o métodos basados en los caracteres (como el método de máxima verosimilitud y el método Bayesiano Markov Chain Monte Carlo). La secuenciación de nueva generación (NGS) se ha vuelto más asequible y se está utilizando cada vez más en la sanidad pública para describir los patógenos causantes de enfermedades infecciosas. Los dispositivos de secuenciación portátiles reducen el tiempo de respuesta y prometen hacer que los datos estén disponibles para apoyar la investigación de brotes en tiempo real. Los datos de NGS pueden utilizarse para identificar el origen o la fuente de una cepa de un brote y su propagación, así como para determinar la presencia de genes de resistencia a los antimicrobianos. Para visualizar el parentesco genético entre las muestras se construye un árbol filogenético.</p>
<p>En esta página aprenderemos a utilizar el paquete <strong>ggtree</strong>, que permite la visualización combinada de árboles filogenéticos con datos de muestra adicionales en forma de dataframe. Esto nos permitirá observar patrones y mejorar la comprensión de la dinámica de los brotes.</p>
<pre><code>## Installing package into '/home/peron/R/x86_64-pc-linux-gnu-library/4.1'
## (as 'lib' is unspecified)</code></pre>
<pre><code>## Warning: package 'ggtree' is not available for this version of R
## 
## A version of this package for your version of R might be available elsewhere,
## see the ideas at
## https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages</code></pre>
<pre><code>## 
## ggtree installed</code></pre>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_overview_graph-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="preparation-29" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Preparación<a class="anchor" aria-label="anchor" href="#preparation-29"><i class="fas fa-link"></i></a>
</h2>
<div id="load-packages-25" class="section level3" number="1.2.1">
<h3>
<span class="header-section-number">1.2.1</span> Cargar paquetes<a class="anchor" aria-label="anchor" href="#load-packages-25"><i class="fas fa-link"></i></a>
</h3>
<p>Este trozo de código muestra la carga de los paquetes necesarios. En este manual destacamos ‘p_load()’ de <strong>pacman</strong>, que instala el paquete si es necesario <em>y</em> lo carga para su uso. También puede cargar los paquetes instalados con library() de <strong>.</strong> Consulta la página sobre <a href="#r-basics">los fundamentos de R</a> para obtener más información sobre los paquetes de R.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span>
  <span class="va">rio</span>,             <span class="co"># import/export</span>
  <span class="va">here</span>,            <span class="co"># relative file paths</span>
  <span class="va">tidyverse</span>,       <span class="co"># general data management and visualization</span>
  <span class="va">ape</span>,             <span class="co"># to import and export phylogenetic files</span>
  <span class="va">ggtree</span>,          <span class="co"># to visualize phylogenetic files</span>
  <span class="va">treeio</span>,          <span class="co"># to visualize phylogenetic files</span>
  <span class="va">ggnewscale</span><span class="op">)</span>      <span class="co"># to add additional layers of color schemes</span></code></pre></div>
</div>
<div id="import-data-23" class="section level3" number="1.2.2">
<h3>
<span class="header-section-number">1.2.2</span> Importar datos<a class="anchor" aria-label="anchor" href="#import-data-23"><i class="fas fa-link"></i></a>
</h3>
<p>Los datos de esta página pueden descargarse con las instrucciones de la página de <a href="#download-handbook-and-data">descarga de manuales y datos</a>.</p>
<p>Hay varios formatos diferentes en los que se puede almacenar un árbol filogenético (por ejemplo, Newick, NEXUS, Phylip). Uno de los más comunes es el formato de archivo Newick (.nwk), que es el estándar para representar árboles en forma legible por ordenador. Esto significa que un árbol completo puede expresarse en un formato de cadena como “((t2:0,04,t1:0,34):0,89,(t5:0,37,(t4:0,03,t3:0,67):0,9):0,59);”, enumerando todos los nodos y puntas y su relación (longitud de rama) entre sí.</p>
<p>Nota: Es importante entender que el archivo del árbol filogenético en sí mismo no contiene datos de secuenciación, sino que es simplemente el resultado de las distancias genéticas entre las secuencias. Por lo tanto, no podemos extraer datos de secuenciación de un archivo de árbol.</p>
<p>En primer lugar, utilizamos la función read.tree() del paquete <strong>ape</strong> para importar un archivo de árbol filogenético de Newick en formato .txt, y lo almacenamos en un objeto tipo lista “phylo”. Si es necesario, utiliza la función <code>here()</code> del paquete <strong>here</strong> para especificar la ruta relativa del archivo.</p>
<p>Nota: En este caso el árbol newick se guarda como un archivo .txt para facilitar su manejo y descarga desde Github.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="st">"Shigella_tree.txt"</span><span class="op">)</span></code></pre></div>
<p>Inspeccionamos nuestro objeto árbol y vemos que contiene 299 puntas (o muestras) y 236 nodos.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 299 tips and 236 internal nodes.
## 
## Tip labels:
##   SRR5006072, SRR4192106, S18BD07865, S18BD00489, S17BD08906, S17BD05939, ...
## Node labels:
##   17, 29, 100, 67, 100, 100, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>En segundo lugar, importamos una tabla almacenada en un archivo .csv con información adicional para cada muestra secuenciada, como el sexo, el país de origen y los atributos de resistencia a los antimicrobianos, utilizando la función <code>import()</code> del paquete <strong>rio</strong>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="st">"sample_data_Shigella_tree.csv"</span><span class="op">)</span></code></pre></div>
<p>A continuación se muestran las primeras 50 filas de los datos:</p>
<div id="htmlwidget-e8748bfa7c6b3db1845e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e8748bfa7c6b3db1845e">{"x":{"filter":"none","vertical":false,"data":[["S17BD05944","S15BD07413","S18BD07247","S19BD07384","S18BD07338","S18BD02657","S19BD06884","S16BD02574","S16BD04069","S17BD08170","S15BD07250","S17BD07935","S17BD00384","S19BD06034","S17BD08948","S17BD04542","S17BD07399","S17BD08233","S19BD07381","S17BD07961","S18BD09176","S19BD07393","S18BD02580","S17BD02526","S17BD06833","S18BD01024","S18BD03411","S18BD09515","S17BD07332","S18BD06598","S17BD02164","S19BD06386","S19BD06283","S19BD06285","S19BD05221","S18BD06749","S17BD02088","S17BD07692","S14BD03573","S17BD01135","S19BD06689","S15BD06969","S18BD01106","S17BD00525","S19BD07036","S18BD06401","S18BD06486","S17BD06426","S17BD05916","S18BD07708"],["Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei"],["Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium"],["Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe"],[null,"Georgia",null,"Morocco",null,null,null,"Thailand",null,null,null,null,null,null,null,"Iran",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Morocco","France","Turkey","Egypt","Tunisia","Congo",null,null,null,null,null,null,null,null,null,null,"Egypt",null,null],["2017","2015","2018","2019","2018","2018","2019","2016","2016","2017","2015","2017","2017","2019","2017","2017","2017","2017","2019","2017","2018","2019","2018","2017","2017","2018","2018","2018","2017","2018","2017","2019","2019","2019","2019","2018","2017","2017","2014","2017","2019","2015","2018","2017","2019","2018","2018","2017","2017","2018"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL"],[null,"Male","Female","Female","Male","Male","Male","Male","Male","Male","Female",null,"Male","Male","Female","Male","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Male",null,"female","Male","Female","Male","Male","Male","Male","Male","Male","Male","Female"],[null,null,null,null,null,"gyrA D87Y","gyrA D87Y","gyrA D87Y","gyrA D87Y",null,null,"gyrA D87Y",null,null,"gyrA D87Y",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA S83L",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA D87Y",null,null,null,"gyrA D87Y",null,null,null,null,null,null,null],[null,null,null,null,null,"mphA","mphA",null,null,null,null,"mphA",null,null,"mphA",null,null,null,null,null,null,null,null,null,null,"mphA","mphA",null,null,null,null,null,null,null,"mphA/ermB",null,null,"mphA",null,null,null,null,"mphA",null,null,null,null,null,null,null],[8,0,16,8,8,64,64,0,0,64,0,64,8,8,64,8,8,8,8,8,8,16,8,8,8,64,64,8,8,8,8,8,8,2,64,8,8,64,0,8,8,0,64,8,4,8,8,8,8,8],[0.25,0,0.03,0.02,0.02,0.5,0.5,0,0,8,0,0.5,0.02,0.02,0.5,0.25,0.02,0.02,0.02,0.02,0.02,0.12,0.02,0.02,0.02,0.5,0.25,0.03,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,1,0,0.02,0.02,0,0.5,0.02,0.02,4,0.02,0.02,0.25,0.02]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>serotype<\/th>\n      <th>Country<\/th>\n      <th>Continent<\/th>\n      <th>Travel_history<\/th>\n      <th>Year<\/th>\n      <th>Belgium<\/th>\n      <th>Source<\/th>\n      <th>Gender<\/th>\n      <th>gyrA_mutations<\/th>\n      <th>macrolide_resistance_genes<\/th>\n      <th>MIC_AZM<\/th>\n      <th>MIC_CIP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[11,12]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="clean-and-inspect" class="section level3" number="1.2.3">
<h3>
<span class="header-section-number">1.2.3</span> Limpiar e inspeccionar<a class="anchor" aria-label="anchor" href="#clean-and-inspect"><i class="fas fa-link"></i></a>
</h3>
<p>Limpiamos e inspeccionamos nuestros datos: Para asignar los datos de muestra correctos al árbol filogenético, los valores de la columna <code>Sample_ID</code> en el dataframe <code>sample_data</code> deben coincidir con los valores de <code>tip.labels</code> en el archivo <code>tree</code>:</p>
<p>Comprobamos el formato de los <code>tip.labels</code> en el archivo de árbol mirando las 6 primeras entradas usando con <code><a href="https://rdrr.io/r/utils/head.html">head()</a></code> de R <strong>base</strong>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span> </code></pre></div>
<pre><code>## [1] "SRR5006072" "SRR4192106" "S18BD07865" "S18BD00489" "S17BD08906" "S17BD05939"</code></pre>
<p>También nos aseguramos de que la primera columna de nuestro dataframe <code>sample_data</code> sea <code>Sample_ID</code>. Miramos los nombres de las columnas de nuestro dataframe utilizando <code><a href="https://rdrr.io/r/base/colnames.html">colnames()</a></code> de R <strong>base</strong>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span>   </code></pre></div>
<pre><code>##  [1] "Sample_ID"                  "serotype"                   "Country"                    "Continent"                  "Travel_history"            
##  [6] "Year"                       "Belgium"                    "Source"                     "Gender"                     "gyrA_mutations"            
## [11] "macrolide_resistance_genes" "MIC_AZM"                    "MIC_CIP"</code></pre>
<p>Miramos los <code>Sample_IDs</code> en el dataframe para asegurarnos de que el formato es el mismo que en el <code>tip.label</code> (por ejemplo, las letras son todas mayúsculas, no hay barras bajas adicionales _ entre las letras y los números, etc.)</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">)</span> <span class="co"># we again inspect only the first 6 using head()</span></code></pre></div>
<pre><code>## [1] "S17BD05944" "S15BD07413" "S18BD07247" "S19BD07384" "S18BD07338" "S18BD02657"</code></pre>
<p>También podemos comparar si todas las muestras están presentes en el archivo <code>tree</code> y viceversa, generando un vector lógico de TRUE o FALSE donde coinciden o no. Estos no se imprimen aquí, para simplificar.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span> <span class="op">%in%</span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>

<span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">%in%</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Podemos utilizar estos vectores para mostrar cualquier ID de muestra que no esté en el árbol (no hay ninguna).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">[</span><span class="op">!</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">%in%</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">]</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>Al inspeccionar podemos ver que el formato de <code>Sample_ID</code> en el dataframe corresponde al formato de los nombres de las muestras en el <code>tip.labels</code>. No es necesario que estén clasificados en el mismo orden para que coincidan.</p>
<p>Estamos listos para ir!</p>
<!-- ======================================================= -->
</div>
</div>
<div id="simple-tree-visualization" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Visualización simple de un árbol<a class="anchor" aria-label="anchor" href="#simple-tree-visualization"><i class="fas fa-link"></i></a>
</h2>
<div id="different-tree-layouts" class="section level3" number="1.3.1">
<h3>
<span class="header-section-number">1.3.1</span> Different tree layouts<a class="anchor" aria-label="anchor" href="#different-tree-layouts"><i class="fas fa-link"></i></a>
</h3>
<p><strong>ggtree</strong> ofrece muchos formatos de presentación diferentes y algunos pueden ser más adecuados para su propósito específico que otros. A continuación se muestran algunas demostraciones. Para otras opciones, consulta este <a href="http://yulab-smu.top/treedata-book/chapter4.html">libro en línea</a>.</p>
<p>A continuación, algunos ejemplos de disposición de árboles:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>                                            <span class="co"># simple linear tree</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,  branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>                   <span class="co"># simple linear tree with all tips aligned</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span>                         <span class="co"># simple circular tree</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="co"># simple circular tree with all tips aligned</span></code></pre></div>
<p><img src="_main_files/figure-html/phylogenetic_trees_example_formats-1.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-2.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-3.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-4.png" width="50%"></p>
</div>
<div id="simple-tree-plus-sample-data" class="section level3" number="1.3.2">
<h3>
<span class="header-section-number">1.3.2</span> Árbol simple con datos de muestra<a class="anchor" aria-label="anchor" href="#simple-tree-plus-sample-data"><i class="fas fa-link"></i></a>
</h3>
<p>El operador <strong>%&lt;+%</strong> se utiliza para conectar el dataframe <code>sample_data</code> con el archivo <code>tree</code>. La anotación más sencilla de su árbol es la adición de los nombres de las muestras en las puntas, así como la coloración de los puntos de las puntas y, si se desea, de las ramas:</p>
<p>Este es un ejemplo de árbol circular:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout <span class="op">=</span> <span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span> <span class="co"># %&lt;+% adds dataframe with sample data to tree</span>
  <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">Belgium</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># color the branches according to a variable in your dataframe</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Sample Origin"</span>,                      <span class="co"># name of your color scheme (will show up in the legend like this)</span>
    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Yes"</span>, <span class="st">"No"</span><span class="op">)</span>,                     <span class="co"># the different options in your variable</span>
    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NRCSS Belgium"</span>, <span class="st">"Other"</span><span class="op">)</span>,        <span class="co"># how you want the different options named in your legend, allows for formatting</span>
    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span><span class="op">)</span>,                  <span class="co"># the color you want to assign to the variable </span>
    na.value <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span>                        <span class="co"># color NA values in black as well</span>
  <span class="fu">new_scale_color</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>                             <span class="co"># allows to add an additional color scheme for another variable</span>
    <span class="fu">geom_tippoint</span><span class="op">(</span>
      mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">Continent</span><span class="op">)</span>,          <span class="co"># tip color by continent. You may change shape adding "shape = "</span>
      size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span>                               <span class="co"># define the size of the point at the tip</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Continent"</span>,                    <span class="co"># name of your color scheme (will show up in the legend like this)</span>
    palette <span class="op">=</span> <span class="st">"Set1"</span>,                      <span class="co"># we choose a set of colors coming with the brewer package</span>
    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span> <span class="op">+</span>                    <span class="co"># for the NA values we choose the color grey</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                             <span class="co"># adds name of sample to tip of its branch </span>
    color <span class="op">=</span> <span class="st">'black'</span>,                       <span class="co"># (add as many text lines as you wish with + , but you may need to adjust offset value to place them next to each other)</span>
    offset <span class="op">=</span> <span class="fl">1</span>,
    size <span class="op">=</span> <span class="fl">1</span>,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span>    
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Phylogenetic tree of Shigella sonnei"</span><span class="op">)</span><span class="op">+</span>       <span class="co"># title of your graph</span>
  <span class="fu">theme</span><span class="op">(</span>
    axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, <span class="co"># removes x-axis title</span>
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, <span class="co"># removes y-axis title</span>
    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>    <span class="co"># defines font size and format of the legend title</span>
      face <span class="op">=</span> <span class="st">"bold"</span>,
      size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,   
    legend.text<span class="op">=</span><span class="fu">element_text</span><span class="op">(</span>       <span class="co"># defines font size and format of the legend text</span>
      face <span class="op">=</span> <span class="st">"bold"</span>,
      size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>      <span class="co"># defines font size and format of the plot title</span>
      size <span class="op">=</span> <span class="fl">12</span>,
      face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,  
    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,     <span class="co"># defines placement of the legend</span>
    legend.box <span class="op">=</span> <span class="st">"vertical"</span>,        <span class="co"># defines placement of the legend</span>
    legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>   </code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_adding_sampledata-1.png" width="672" style="display: block; margin: auto;"></div>
<p>Puedes exportar el gráfico de árbol con <code>ggsave()</code> como lo harías con cualquier otro objeto ggplot. Escrito de esta manera, ggsave() guarda la última imagen producida en la ruta de archivo que especifiques. Recuerde que puede utilizar <code>here()</code> y rutas de archivo relativas para guardar fácilmente en subcarpetas, etc.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggsave</span><span class="op">(</span><span class="st">"example_tree_circular_1.png"</span>, width <span class="op">=</span> <span class="fl">12</span>, height <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></code></pre></div>
<!-- ======================================================= -->
</div>
</div>
<div id="tree-manipulation" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Manipulación de árboles<a class="anchor" aria-label="anchor" href="#tree-manipulation"><i class="fas fa-link"></i></a>
</h2>
<p>A veces puede tener un árbol filogenético muy grande y sólo le interesa una parte del árbol. Por ejemplo, si ha producido un árbol que incluye muestras históricas o internacionales para obtener una gran visión general de dónde puede encajar su conjunto de datos en el panorama general. Pero luego, para ver más de cerca sus datos, quiere inspeccionar sólo esa parte del árbol más grande.</p>
<p>Dado que el archivo del árbol filogenético es sólo el resultado del análisis de los datos de secuenciación, no podemos manipular el orden de los nodos y las ramas en el propio archivo. Estos ya han sido determinados en análisis anteriores a partir de los datos NGS en bruto. Sin embargo, podemos ampliar partes, ocultar partes e incluso subconjuntar partes del árbol.</p>
<div id="zoom-in" class="section level3" number="1.4.1">
<h3>
<span class="header-section-number">1.4.1</span> Acercarse<a class="anchor" aria-label="anchor" href="#zoom-in"><i class="fas fa-link"></i></a>
</h3>
<p>Si no quieres “cortar” tu árbol, sino sólo inspeccionar una parte de él más de cerca, puedes acercarte para ver una parte específica.</p>
<p>En primer lugar, trazamos todo el árbol en formato lineal y añadimos etiquetas numéricas a cada nodo del árbol.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,<span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>                <span class="co"># labels the tips of all branches with the sample name in the tree file</span>
  <span class="fu">geom_text2</span><span class="op">(</span>
    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>,
                  label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,
    size <span class="op">=</span> <span class="fl">5</span>,
    color <span class="op">=</span> <span class="st">"darkred"</span>,
    hjust <span class="op">=</span> <span class="fl">1</span>,
    vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>                            <span class="co"># labels all the nodes in the tree</span>

<span class="va">p</span>  <span class="co"># print</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Para acercarse a una rama en particular (que sobresale a la derecha), utiliza <code>viewClade()</code> en el objeto ggtree <code>p</code> y proporcione el número de nodo para verlo más de cerca:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">viewClade</span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in_452-1.png" width="50%" style="display: block; margin: auto;"></div>
</div>
<div id="collapsing-branches" class="section level3" number="1.4.2">
<h3>
<span class="header-section-number">1.4.2</span> Ramas colapsadas<a class="anchor" aria-label="anchor" href="#collapsing-branches"><i class="fas fa-link"></i></a>
</h3>
<p>Sin embargo, podemos querer ignorar esta rama y podemos colapsarla en ese mismo nodo (nodo 452) utilizando <code>collapse()</code>. Este árbol se define como <code>p_collapsed</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_collapsed</span> <span class="op">&lt;-</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span>
<span class="va">p_collapsed</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_collapse_452-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Para mayor claridad, cuando imprimimos p_collapsed, añadimos un geom_point2() (un diamante azul) en el nodo de la rama colapsada.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_collapsed</span> <span class="op">+</span> 
<span class="fu">geom_point2</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>subset <span class="op">=</span> <span class="op">(</span><span class="va">node</span> <span class="op">==</span> <span class="fl">452</span><span class="op">)</span><span class="op">)</span>,  <span class="co"># we assign a symbol to the collapsed node</span>
            size <span class="op">=</span> <span class="fl">5</span>,                     <span class="co"># define the size of the symbol</span>
            shape <span class="op">=</span> <span class="fl">23</span>,                   <span class="co"># define the shape of the symbol</span>
            fill <span class="op">=</span> <span class="st">"steelblue"</span><span class="op">)</span>           <span class="co"># define the color of the symbol</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="672"></div>
</div>
<div id="subsetting-a-tree" class="section level3" number="1.4.3">
<h3>
<span class="header-section-number">1.4.3</span> Subconjunto de un árbol<a class="anchor" aria-label="anchor" href="#subsetting-a-tree"><i class="fas fa-link"></i></a>
</h3>
<p>Si queremos hacer un cambio más permanente y crear un nuevo árbol reducido con el que trabajar, podemos subconjuntar parte de él con <code>tree_subset()</code>. Luego se puede guardar como un nuevo archivo de árbol o archivo .txt.</p>
<p>En primer lugar, inspeccionamos los nodos del árbol y las etiquetas de las puntas para decidir qué se va a subconjuntar.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span>
  <span class="va">tree</span>,
  branch.length <span class="op">=</span> <span class="st">'none'</span>,
  layout <span class="op">=</span> <span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>               <span class="co"># we add the asmple data using the %&lt;+% operator</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">+</span>                                <span class="co"># label tips of all branches with sample name in tree file</span>
  <span class="fu">geom_text2</span><span class="op">(</span>
    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>, label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    color <span class="op">=</span> <span class="st">"darkred"</span><span class="op">)</span> <span class="op">+</span>                                <span class="co"># labels all the nodes in the tree</span>
 <span class="fu">theme</span><span class="op">(</span>
   legend.position <span class="op">=</span> <span class="st">"none"</span>,                            <span class="co"># removes the legend all together</span>
   axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
   axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
   plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_subsetting-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Ahora, digamos que hemos decidido hacer un subconjunto del árbol en el nodo 528 (mantener sólo las puntas dentro de esta rama después del nodo 528) y lo guardamos como un nuevo objeto <code>sub_tree1</code>:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sub_tree1</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span>
  <span class="va">tree</span>,
  node <span class="op">=</span> <span class="fl">528</span><span class="op">)</span>                                            <span class="co"># we subset the tree at node 528</span></code></pre></div>
<p>Veamos el subconjunto del árbol 1:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Subset tree 1"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672"></div>
<p>También puede hacer un subconjunto basado en una muestra en particular, especificando cuántos nodos “hacia atrás” quiere incluir. Vamos a subconjuntar la misma parte del árbol basándonos en una muestra, en este caso S17BD07692, retrocediendo 9 nodos y lo guardamos como un nuevo objeto <code>sub_tree2</code>:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sub_tree2</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span>
  <span class="va">tree</span>,
  <span class="st">"S17BD07692"</span>,
  levels_back <span class="op">=</span> <span class="fl">9</span><span class="op">)</span> <span class="co"># levels back defines how many nodes backwards from the sample tip you want to go</span></code></pre></div>
<p>Veamos el subconjunto del árbol 2:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span>  <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Subset tree 2"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-20-1.png" width="672"></div>
<p>También puede guardar su nuevo árbol como un tipo Newick o incluso un archivo de texto utilizando la función <code>write.tree()</code> del paquete <strong>ape</strong>:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># to save in .nwk format</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.nwk'</span><span class="op">)</span>

<span class="co"># to save in .txt format</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.txt'</span><span class="op">)</span></code></pre></div>
</div>
<div id="rotating-nodes-in-a-tree" class="section level3" number="1.4.4">
<h3>
<span class="header-section-number">1.4.4</span> Rotación de nodos en un árbol<a class="anchor" aria-label="anchor" href="#rotating-nodes-in-a-tree"><i class="fas fa-link"></i></a>
</h3>
<p>Como ya hemos dicho, no podemos cambiar el orden de las puntas o de los nodos en el árbol, ya que éste se basa en su parentesco genético y no está sujeto a manipulación visual. Pero podemos rotar las ramas alrededor de los nodos si eso facilita nuestra visualización.</p>
<p>En primer lugar, trazamos nuestro nuevo subconjunto de árbol 2 con las etiquetas de los nodos para elegir el nodo que queremos manipular y lo almacenamos en un objeto ggtree plot <code>p</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text2</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>subset<span class="op">=</span><span class="op">!</span><span class="va">isTip</span>, label<span class="op">=</span><span class="va">node</span><span class="op">)</span>, <span class="co"># labels all the nodes in the tree</span>
             size <span class="op">=</span> <span class="fl">5</span>,
             color <span class="op">=</span> <span class="st">"darkred"</span>, 
             hjust <span class="op">=</span> <span class="fl">1</span>, 
             vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> 
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_1-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Luego podemos manipular los nodos aplicando ggtree<strong>::rotate()</strong> o <strong>ggtree::flip()</strong>: Nota: para ilustrar qué nodos estamos manipulando aplicamos primero la función <strong>geom_hilight()</strong> de <strong>ggtree</strong> para resaltar las muestras de los nodos que nos interesan y almacenamos ese objeto de trazado de ggtree en un nuevo objeto <code>p1</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu">geom_hilight</span><span class="op">(</span>  <span class="co"># highlights node 39 in blue, "extend =" allows us to define the length of the color block</span>
  node <span class="op">=</span> <span class="fl">39</span>,
  fill <span class="op">=</span> <span class="st">"steelblue"</span>,
  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>  
<span class="fu">geom_hilight</span><span class="op">(</span>            <span class="co"># highlights the node 37 in yellow</span>
  node <span class="op">=</span> <span class="fl">37</span>,
  fill <span class="op">=</span> <span class="st">"yellow"</span>,
  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>               
<span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Original tree"</span><span class="op">)</span>


<span class="va">p1</span> <span class="co"># print</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_2-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Ahora podemos rotar el nodo 37 en el objeto <code>p1</code> para que las muestras del nodo 38 se muevan hacia arriba. Almacenamos el árbol rotado en un nuevo objeto p2</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">rotate</span><span class="op">(</span><span class="va">p1</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span> 
      <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Rotated Node 37"</span><span class="op">)</span>


<span class="va">p2</span>   <span class="co"># print</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-21-1.png" width="672"></div>
<p>O podemos usar el comando flip para rotar el nodo 36 en el objeto p1 y cambiar el nodo 37 a la parte superior y el nodo 39 a la parte inferior. Almacenamos el árbol volteado en un nuevo objeto <code>p3</code>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p3</span> <span class="op">&lt;-</span>  <span class="fu">flip</span><span class="op">(</span><span class="va">p1</span>, <span class="fl">39</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Rotated Node 36"</span><span class="op">)</span>


<span class="va">p3</span>   <span class="co"># print</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-22-1.png" width="672"></div>
</div>
<div id="example-subtree-with-sample-data-annotation" class="section level3" number="1.4.5">
<h3>
<span class="header-section-number">1.4.5</span> Ejemplo de subárbol con anotación de datos de muestra<a class="anchor" aria-label="anchor" href="#example-subtree-with-sample-data-annotation"><i class="fas fa-link"></i></a>
</h3>
<p>Digamos que estamos investigando el grupo de casos con expansión clonal que se produjo en 2017 y 2018 en el nodo 39 de nuestro subárbol. Añadimos el año de aislamiento de la cepa, así como el historial de viajes y el color por país para ver el origen de otras cepas estrechamente relacionadas:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>     <span class="co"># we use th %&lt;+% operator to link to the sample_data</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># labels the tips of all branches with the sample name in the tree file</span>
    size <span class="op">=</span> <span class="fl">2.5</span>,
    offset <span class="op">=</span> <span class="fl">0.001</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># set the x-axis limits of our tree</span>
  <span class="fu">geom_tippoint</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">Country</span><span class="op">)</span>,     <span class="co"># color the tip point by continent</span>
                size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">scale_color_brewer</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Country"</span>, 
    palette <span class="op">=</span> <span class="st">"Set1"</span>, 
    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># add isolation year as a text label at the tips</span>
    <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">Year</span><span class="op">)</span>,
    color <span class="op">=</span> <span class="st">'blue'</span>,
    offset <span class="op">=</span> <span class="fl">0.0045</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    linetype <span class="op">=</span> <span class="st">"blank"</span> ,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># add travel history as a text label at the tips, in red color</span>
    <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">Travel_history</span><span class="op">)</span>,
    color <span class="op">=</span> <span class="st">'red'</span>,
    offset <span class="op">=</span> <span class="fl">0.006</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    linetype <span class="op">=</span> <span class="st">"blank"</span>,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Phylogenetic tree of Belgian S. sonnei strains with travel history"</span><span class="op">)</span><span class="op">+</span>  <span class="co"># add plot title</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"genetic distance (0.001 = 4 nucleotides difference)"</span><span class="op">)</span><span class="op">+</span>                    <span class="co"># add a label to the x-axis </span>
  <span class="fu">theme</span><span class="op">(</span>
    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_inspect_subset_example-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Nuestra observación apunta a un evento de importación de cepas procedentes de Asia, que luego circularon en Bélgica a lo largo de los años y parecen haber causado nuestro último brote.</p>
<!-- ======================================================= -->
</div>
</div>
<div id="more-complex-trees-adding-heatmaps-of-sample-data" class="section level2" number="1.5">
<h2>
<span class="header-section-number">1.5</span> Árboles más complejos: añadir mapas térmicos de datos de muestra<a class="anchor" aria-label="anchor" href="#more-complex-trees-adding-heatmaps-of-sample-data"><i class="fas fa-link"></i></a>
</h2>
<p>Podemos añadir información más compleja, como la presencia categórica de genes de resistencia a los antimicrobianos y valores numéricos para la resistencia realmente medida a los antimicrobianos en forma de mapa de calor utilizando la función <strong>ggtree::gheatmap()</strong>.</p>
<p>Primero necesitamos trazar nuestro árbol (puede ser lineal o circular) y almacenarlo en un nuevo objeto ggtree plot <code>p</code>: Utilizaremos el subárbol de la parte 3).</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span>, branch.length<span class="op">=</span><span class="st">'none'</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span> 
 <span class="fu">theme</span><span class="op">(</span>
   legend.position <span class="op">=</span> <span class="st">"none"</span>,
    axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>
      size <span class="op">=</span> <span class="fl">12</span>,
      face <span class="op">=</span> <span class="st">"bold"</span>,
      hjust <span class="op">=</span> <span class="fl">0.5</span>,
      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap-1.png" width="60%" style="display: block; margin: auto;"></div>
<p>En segundo lugar, preparamos nuestros datos. Para visualizar las diferentes variables con nuevos esquemas de color, subestablecimos nuestro dataframe a la variable deseada. Es importante añadir el <code>Sample_ID</code> como rownames de lo contrario no puede coincidir los datos con el árbol <code>tip.labels</code>:</p>
<p>En nuestro ejemplo, queremos observar el género y las mutaciones que podrían conferir resistencia a la ciprofloxacina, un importante antibiótico de primera línea utilizado para tratar las infecciones por Shigella.</p>
<p>Creamos un dataframe para el género:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"gender"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gender"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gender</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un dataframe para las mutaciones en el gen gyrA, que confieren resistencia a la ciprofloxacina:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cipR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"cipR"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gyrA_mutations"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cipR</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un dataframe para la concentración inhibitoria mínima (CIM) medida para la Ciprofloxacina del laboratorio:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MIC_Cip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"mic_cip"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MIC_CIP"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">MIC_Cip</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un primer gráfico añadiendo un mapa de calor binario para el género al árbol filogenético y almacenándolo en un nuevo objeto de gráfico ggtree <code>h1</code>:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>,                                 <span class="co"># we add a heatmap layer of the gender dataframe to our tree plot</span>
                offset <span class="op">=</span> <span class="fl">10</span>,                               <span class="co"># offset shifts the heatmap to the right,</span>
                width <span class="op">=</span> <span class="fl">0.10</span>,                              <span class="co"># width defines the width of the heatmap column,</span>
                color <span class="op">=</span> <span class="cn">NULL</span>,                              <span class="co"># color defines the boarder of the heatmap columns</span>
         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>                               <span class="co"># hides column names for the heatmap</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,                       <span class="co"># define the coloring scheme and legend for gender</span>
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_gender-1.png" width="70%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos información sobre las mutaciones en el gen gyrA, que confieren resistencia a la ciprofloxacina:</p>
<p>Nota: La presencia de mutaciones puntuales cromosómicas en los datos de WGS se determinó previamente utilizando la herramienta PointFinder desarrollada por Zankari et al. (véase la referencia en la sección de referencias adicionales)</p>
<p>En primer lugar, asignamos un nuevo esquema de colores a nuestro objeto de trazado existente <code>h1</code> y lo almacenamos en un objeto ahora <code>h2</code>. Esto nos permite definir y cambiar los colores para nuestra segunda variable en el mapa de calor.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>A continuación, añadimos la segunda capa del mapa de calor a <code>h2</code> y almacenamos los gráficos combinados en un nuevo objeto <code>h3</code>:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,         <span class="co"># adds the second row of heatmap describing Ciprofloxacin resistance mutations</span>
               offset <span class="op">=</span> <span class="fl">12</span>, 
               width <span class="op">=</span> <span class="fl">0.10</span>, 
               colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Ciprofloxacin resistance \n conferring mutation"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h3</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_genes-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Repetimos el proceso anterior, añadiendo primero una nueva capa de escala de colores a nuestro objeto existente <code>h3</code>, y luego añadiendo los datos continuos sobre la concentración inhibitoria mínima (CIM) de Ciprofloxacina para cada cepa al objeto resultante <code>h4</code> para producir el objeto final <code>h5</code>:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First we add the new coloring scheme:</span>
<span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>

<span class="co"># then we combine the two into a new plot:</span>
<span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>,  
               offset <span class="op">=</span> <span class="fl">14</span>, 
               width <span class="op">=</span> <span class="fl">0.10</span>,
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC for Ciprofloxacin"</span>,  <span class="co"># here we define a gradient color scheme for the continuous variable of MIC</span>
                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,
                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.50</span>, <span class="fl">1.00</span><span class="op">)</span>,
                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colourbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_MIC-1.png" width="90%" style="display: block; margin: auto;"></div>
<p>Podemos hacer el mismo ejercicio para un árbol lineal:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="co"># labels the tips</span>
  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"genetic distance (0.001 = 4 nucleotides difference)"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,
      axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, 
                                face <span class="op">=</span> <span class="st">"bold"</span>,
                                hjust <span class="op">=</span> <span class="fl">0.5</span>,
                                vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_1-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Primero añadimos el género:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>, 
                offset <span class="op">=</span> <span class="fl">0.003</span>,
                width <span class="op">=</span> <span class="fl">0.1</span>, 
                color<span class="op">=</span><span class="st">"black"</span>, 
         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_2-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos las mutaciones de resistencia a la ciprofloxacina después de añadir otra capa de colores:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>
<span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,   
               offset <span class="op">=</span> <span class="fl">0.004</span>, 
               width <span class="op">=</span> <span class="fl">0.1</span>,
               color <span class="op">=</span> <span class="st">"black"</span>,
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Ciprofloxacin resistance \n conferring mutation"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="va">h3</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_3-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos la concentración mínima inhibitoria determinada por el laboratorio (MIC):</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>
<span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>, 
               offset <span class="op">=</span> <span class="fl">0.005</span>,  
               width <span class="op">=</span> <span class="fl">0.1</span>,
               color <span class="op">=</span> <span class="st">"black"</span>, 
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC for Ciprofloxacin"</span>,
                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,
                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.50</span>,<span class="fl">1.00</span><span class="op">)</span>,
                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colourbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"horizontal"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>shape <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_4-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="recursos-resources-29" class="section level2" number="1.6">
<h2>
<span class="header-section-number">1.6</span> Recursos {resources-29}<a class="anchor" aria-label="anchor" href="#recursos-resources-29"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#" class="uri">http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#</a> Clade_Colors
<a href="https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a>
<a href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html" class="uri">https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html</a>
<a href="https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a></p>
<p>Ea Zankari, Rosa Allesøe, Katrine G Joensen, Lina M Cavaco, Ole Lund, Frank M Aarestrup, PointFinder: una novedosa herramienta web para la detección basada en WGS de la resistencia a los antimicrobianos asociada a mutaciones puntuales cromosómicas en patógenos bacterianos, Journal of Antimicrobial Chemotherapy, Volume 72, Issue 10, October 2017, Pages 2764-2768, <a href="https://doi.org/10.1093/jac/dkx217">https://doi.org/10.1093/jac/dkx217</a></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html"></a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#phylogenetic-trees-1"><span class="header-section-number">1</span> Árboles filogenéticos. Borrador</a></li>
<li><a class="nav-link" href="#overview-7"><span class="header-section-number">1.1</span> Resumen</a></li>
<li>
<a class="nav-link" href="#preparation-29"><span class="header-section-number">1.2</span> Preparación</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#load-packages-25"><span class="header-section-number">1.2.1</span> Cargar paquetes</a></li>
<li><a class="nav-link" href="#import-data-23"><span class="header-section-number">1.2.2</span> Importar datos</a></li>
<li><a class="nav-link" href="#clean-and-inspect"><span class="header-section-number">1.2.3</span> Limpiar e inspeccionar</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#simple-tree-visualization"><span class="header-section-number">1.3</span> Visualización simple de un árbol</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#different-tree-layouts"><span class="header-section-number">1.3.1</span> Different tree layouts</a></li>
<li><a class="nav-link" href="#simple-tree-plus-sample-data"><span class="header-section-number">1.3.2</span> Árbol simple con datos de muestra</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#tree-manipulation"><span class="header-section-number">1.4</span> Manipulación de árboles</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#zoom-in"><span class="header-section-number">1.4.1</span> Acercarse</a></li>
<li><a class="nav-link" href="#collapsing-branches"><span class="header-section-number">1.4.2</span> Ramas colapsadas</a></li>
<li><a class="nav-link" href="#subsetting-a-tree"><span class="header-section-number">1.4.3</span> Subconjunto de un árbol</a></li>
<li><a class="nav-link" href="#rotating-nodes-in-a-tree"><span class="header-section-number">1.4.4</span> Rotación de nodos en un árbol</a></li>
<li><a class="nav-link" href="#example-subtree-with-sample-data-annotation"><span class="header-section-number">1.4.5</span> Ejemplo de subárbol con anotación de datos de muestra</a></li>
</ul>
</li>
<li><a class="nav-link" href="#more-complex-trees-adding-heatmaps-of-sample-data"><span class="header-section-number">1.5</span> Árboles más complejos: añadir mapas térmicos de datos de muestra</a></li>
<li><a class="nav-link" href="#recursos-resources-29"><span class="header-section-number">1.6</span> Recursos {resources-29}</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>El manual de R para la Epidemiología</strong>" was written by El equipo del manual. It was last built on 2021-10-26.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
