<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>38 Árboles filogenéticos | EpiRhandbook en español</title>
<meta name="author" content="El equipo del manual">
<meta name="description" content="38.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<meta name="generator" content="bookdown 0.29 with bs4_book()">
<meta property="og:title" content="38 Árboles filogenéticos | EpiRhandbook en español">
<meta property="og:type" content="book">
<meta property="og:description" content="38.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="38 Árboles filogenéticos | EpiRhandbook en español">
<meta name="twitter:description" content="38.1 Resumen Los árboles filogenéticos se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético. Pueden construirse a...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.0/transition.js"></script><script src="libs/bs3compat-0.4.0/tabs.js"></script><script src="libs/bs3compat-0.4.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.25/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/tabwid-1.1.0/tabwid.css" rel="stylesheet">
<link href="libs/tabwid-1.1.0/scrool.css" rel="stylesheet">
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><script src="libs/d3-4.5.0/d3.min.js"></script><script src="libs/sankey-1/sankey.js"></script><script src="libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><script src="libs/plotly-binding-4.10.0/plotly.js"></script><script src="libs/typedarray-0.1/typedarray.min.js"></script><link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script><link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="libs/vis-9.1.0/vis-network.min.js"></script><script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script><link href="libs/font-awesome-4.7.0/css/font-awesome.min.css" rel="stylesheet">
<script src="libs/plotly-basic-2.5.1/plotly-basic-2.5.1.min.js"></script><script src="libs/plotly-cartesian-2.5.1/plotly-cartesian-2.5.1.min.js"></script><script src="libs/proj4js-2.3.15/proj4.js"></script><link href="libs/highcharts-9.3.1/css/motion.css" rel="stylesheet">
<script src="libs/highcharts-9.3.1/highcharts.js"></script><script src="libs/highcharts-9.3.1/highcharts-3d.js"></script><script src="libs/highcharts-9.3.1/highcharts-more.js"></script><script src="libs/highcharts-9.3.1/modules/stock.js"></script><script src="libs/highcharts-9.3.1/modules/map.js"></script><script src="libs/highcharts-9.3.1/modules/data.js"></script><script src="libs/highcharts-9.3.1/modules/exporting.js"></script><script src="libs/highcharts-9.3.1/modules/offline-exporting.js"></script><script src="libs/highcharts-9.3.1/modules/drilldown.js"></script><script src="libs/highcharts-9.3.1/modules/item-series.js"></script><script src="libs/highcharts-9.3.1/modules/overlapping-datalabels.js"></script><script src="libs/highcharts-9.3.1/modules/annotations.js"></script><script src="libs/highcharts-9.3.1/modules/export-data.js"></script><script src="libs/highcharts-9.3.1/modules/funnel.js"></script><script src="libs/highcharts-9.3.1/modules/heatmap.js"></script><script src="libs/highcharts-9.3.1/modules/treemap.js"></script><script src="libs/highcharts-9.3.1/modules/sankey.js"></script><script src="libs/highcharts-9.3.1/modules/dependency-wheel.js"></script><script src="libs/highcharts-9.3.1/modules/organization.js"></script><script src="libs/highcharts-9.3.1/modules/solid-gauge.js"></script><script src="libs/highcharts-9.3.1/modules/streamgraph.js"></script><script src="libs/highcharts-9.3.1/modules/sunburst.js"></script><script src="libs/highcharts-9.3.1/modules/vector.js"></script><script src="libs/highcharts-9.3.1/modules/wordcloud.js"></script><script src="libs/highcharts-9.3.1/modules/xrange.js"></script><script src="libs/highcharts-9.3.1/modules/tilemap.js"></script><script src="libs/highcharts-9.3.1/modules/venn.js"></script><script src="libs/highcharts-9.3.1/modules/gantt.js"></script><script src="libs/highcharts-9.3.1/modules/timeline.js"></script><script src="libs/highcharts-9.3.1/modules/parallel-coordinates.js"></script><script src="libs/highcharts-9.3.1/modules/bullet.js"></script><script src="libs/highcharts-9.3.1/modules/coloraxis.js"></script><script src="libs/highcharts-9.3.1/modules/dumbbell.js"></script><script src="libs/highcharts-9.3.1/modules/lollipop.js"></script><script src="libs/highcharts-9.3.1/modules/series-label.js"></script><script src="libs/highcharts-9.3.1/plugins/motion.js"></script><script src="libs/highcharts-9.3.1/custom/reset.js"></script><script src="libs/highcharts-9.3.1/modules/boost.js"></script><script src="libs/highchart-binding-0.9.4/highchart.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style_bs4.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">EpiRhandbook en español</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Buscar" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Tabla de contenidos</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"></a></li>
<li class="book-part">Acerca de este libro</li>
<li><a class="" href="editorial-and-technical-notes.html"><span class="header-section-number">1</span> Notas editoriales y técnicas</a></li>
<li><a class="" href="download-handbook-and-data.html"><span class="header-section-number">2</span> Descargando el manual y los datos</a></li>
<li class="book-part">Aspectos básicos</li>
<li><a class="" href="r-basics.html"><span class="header-section-number">3</span> Fundamentos de R</a></li>
<li><a class="" href="transition-to-r.html"><span class="header-section-number">4</span> Transición a R</a></li>
<li><a class="" href="suggested-packages-1.html"><span class="header-section-number">5</span> Paquetes recomendados</a></li>
<li><a class="" href="r-projects.html"><span class="header-section-number">6</span> Proyectos en R</a></li>
<li><a class="" href="import-and-export.html"><span class="header-section-number">7</span> Importación y exportación</a></li>
<li class="book-part">Gestión de datos</li>
<li><a class="" href="cleaning-data-and-core-functions.html"><span class="header-section-number">8</span> Limpieza de datos y funciones básicas</a></li>
<li><a class="" href="working-with-dates.html"><span class="header-section-number">9</span> Trabajando con Fechas</a></li>
<li><a class="" href="characters-and-strings.html"><span class="header-section-number">10</span> Caracteres y cadenas</a></li>
<li><a class="" href="factors.html"><span class="header-section-number">11</span> Factores</a></li>
<li><a class="" href="pivoting-data.html"><span class="header-section-number">12</span> Pivotar datos</a></li>
<li><a class="" href="grouping-data.html"><span class="header-section-number">13</span> Agrupar datos</a></li>
<li><a class="" href="joining-data.html"><span class="header-section-number">14</span> Unir datos</a></li>
<li><a class="" href="de-duplication.html"><span class="header-section-number">15</span> De-duplicación</a></li>
<li><a class="" href="iteration-loops-and-lists.html"><span class="header-section-number">16</span> Iteración, bucles y listas</a></li>
<li class="book-part">Análisis</li>
<li><a class="" href="descriptive-tables.html"><span class="header-section-number">17</span> Tablas descriptivas</a></li>
<li><a class="" href="simple-statistical-tests.html"><span class="header-section-number">18</span> Tests estadísticos sencillos</a></li>
<li><a class="" href="univariate-and-multivariable-regression.html"><span class="header-section-number">19</span> Regresión univariante y multivariable</a></li>
<li><a class="" href="missing-data.html"><span class="header-section-number">20</span> Valores faltantes</a></li>
<li><a class="" href="standardised-rates.html"><span class="header-section-number">21</span> Tasas estandarizadas</a></li>
<li><a class="" href="moving-averages.html"><span class="header-section-number">22</span> Medias Móviles</a></li>
<li><a class="" href="time-series-and-outbreak-detection.html"><span class="header-section-number">23</span> Series temporales y detección de brotes</a></li>
<li><a class="" href="epidemic-modeling.html"><span class="header-section-number">24</span> Modelización de epidemias</a></li>
<li><a class="" href="contact-tracing-1.html"><span class="header-section-number">25</span> Rastreo de contactos</a></li>
<li><a class="" href="survey-analysis.html"><span class="header-section-number">26</span> Análisis de encuestas</a></li>
<li><a class="" href="survival-analysis.html"><span class="header-section-number">27</span> Análisis de supervivencia</a></li>
<li><a class="" href="gis-basics.html"><span class="header-section-number">28</span> Conceptos básicos de los SIG</a></li>
<li class="book-part">Visualización de datos</li>
<li><a class="" href="tables-for-presentation.html"><span class="header-section-number">29</span> Tablas para presentaciones</a></li>
<li><a class="" href="ggplot-basics.html"><span class="header-section-number">30</span> Conceptos básicos de ggplot</a></li>
<li><a class="" href="ggplot-tips.html"><span class="header-section-number">31</span> Consejos de ggplot</a></li>
<li><a class="" href="epidemic-curves.html"><span class="header-section-number">32</span> Curvas epidémicas</a></li>
<li><a class="" href="demographic-pyramids-and-likert-scales.html"><span class="header-section-number">33</span> Pirámides de población y escalas de Likert</a></li>
<li><a class="" href="heat-plots.html"><span class="header-section-number">34</span> Gráficos de calor</a></li>
<li><a class="" href="diagrams-and-charts.html"><span class="header-section-number">35</span> Diagramas y gráficos</a></li>
<li><a class="" href="combinations-analysis.html"><span class="header-section-number">36</span> Análisis de combinaciones</a></li>
<li><a class="" href="transmission-chains.html"><span class="header-section-number">37</span> Cadenas de transmisión</a></li>
<li><a class="active" href="phylogenetic-trees-1.html"><span class="header-section-number">38</span> Árboles filogenéticos</a></li>
<li><a class="" href="interactive-plots.html"><span class="header-section-number">39</span> Gráficos interactivos</a></li>
<li class="book-part">Informes y Dashboards</li>
<li><a class="" href="reports-with-r-markdown.html"><span class="header-section-number">40</span> Informes con R Markdown</a></li>
<li><a class="" href="organizing-routine-reports.html"><span class="header-section-number">41</span> Organización de informes rutinarios</a></li>
<li><a class="" href="dashboards-with-r-markdown.html"><span class="header-section-number">42</span> Dashboards con R Markdown</a></li>
<li><a class="" href="dashboards-with-shiny.html"><span class="header-section-number">43</span> Dashboards con Shiny</a></li>
<li class="book-part">Miscelánea</li>
<li><a class="" href="writing-functions-1.html"><span class="header-section-number">44</span> Escribir funciones</a></li>
<li><a class="" href="directory-interactions.html"><span class="header-section-number">45</span> Interacciones con directorios</a></li>
<li><a class="" href="version-control-and-collaboration-with-git-and-github.html"><span class="header-section-number">46</span> Control de versiones y colaboración con Git y Github</a></li>
<li><a class="" href="common-errors.html"><span class="header-section-number">47</span> Errores comunes</a></li>
<li><a class="" href="getting-help.html"><span class="header-section-number">48</span> Cómo obtener ayuda</a></li>
<li><a class="" href="r-on-network-drives.html"><span class="header-section-number">49</span> R en redes locales</a></li>
<li><a class="" href="data-table.html"><span class="header-section-number">50</span> Data Table</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="phylogenetic-trees-1" class="section level1" number="38">
<h1>
<span class="header-section-number">38</span> Árboles filogenéticos<a class="anchor" aria-label="anchor" href="#phylogenetic-trees-1"><i class="fas fa-link"></i></a>
</h1>
<!-- ======================================================= -->
<div id="overview-8" class="section level2" number="38.1">
<h2>
<span class="header-section-number">38.1</span> Resumen<a class="anchor" aria-label="anchor" href="#overview-8"><i class="fas fa-link"></i></a>
</h2>
<p>Los <strong>árboles filogenéticos</strong> se utilizan para visualizar y describir el parentesco y la evolución de los organismos a partir de la secuencia de su código genético.</p>
<p>Pueden construirse a partir de secuencias genéticas utilizando métodos basados en la distancia (como el método de unión de vecinos) o métodos basados en los caracteres (como el método de máxima verosimilitud y el método Bayesiano Markov Chain Monte Carlo). La secuenciación de nueva generación (NGS, por sus siglas en inglés) se ha vuelto más económica y se está utilizando cada vez más en el área de salud pública para describir los patógenos causantes de enfermedades infecciosas. Los dispositivos de secuenciación portátil reducen el tiempo de respuesta y prometen facilitar los datos en tiempo real y así apoyar la investigación de brotes. Los datos de NGS se pueden utilizar para identificar el origen o la fuente de una cepa de un brote y su propagación, así como para determinar la presencia de genes de resistencia antimicrobiana. Para visualizar el parentesco genético entre muestras biológicas se construye un árbol filogenético.</p>
<p>Aquí aprenderemos a utilizar el paquete <strong>ggtree</strong>, que permite la visualización combinada de árboles filogenéticos con datos de muestra adicionales en forma de dataframe. Esto nos permitirá observar patrones y comprender mejor la dinámica de los brotes.</p>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_overview_graph-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="preparation-31" class="section level2" number="38.2">
<h2>
<span class="header-section-number">38.2</span> Preparación<a class="anchor" aria-label="anchor" href="#preparation-31"><i class="fas fa-link"></i></a>
</h2>
<div id="cargar-paquetes-27" class="section level3 unnumbered">
<h3>Cargar paquetes<a class="anchor" aria-label="anchor" href="#cargar-paquetes-27"><i class="fas fa-link"></i></a>
</h3>
<p>Este trozo de código muestra la carga de los paquetes necesarios. En este manual destacamos ‘p_load()’ de <strong>pacman</strong>, que instala el paquete si es necesario <em>y</em> lo carga para su uso. También puede cargar los paquetes instalados con library() de R <strong>base</strong>. Consulta la página <a href="r-basics.html#r-basics">Fundamentos de R</a> para obtener más información sobre los paquetes de R.</p>
<div class="sourceCode" id="cb1735"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span>
  <span class="va">rio</span>,             <span class="co"># importa/exportar</span>
  <span class="va">here</span>,            <span class="co"># rutas de fichero relativas</span>
  <span class="va">tidyverse</span>,       <span class="co"># manejo general de datos y visualización</span>
  <span class="va">ape</span>,             <span class="co"># para importar y exportar archivos filogenéticos</span>
  <span class="va">ggtree</span>,          <span class="co"># para visualizar archivos filogenéticos</span>
  <span class="va">treeio</span>,          <span class="co"># para visualizar archivos filogenéticos</span>
  <span class="va">ggnewscale</span><span class="op">)</span>      <span class="co"># para añadir capas adicionales de esquemas de color</span></code></pre></div>
</div>
<div id="importar-datos-22" class="section level3 unnumbered">
<h3>Importar datos<a class="anchor" aria-label="anchor" href="#importar-datos-22"><i class="fas fa-link"></i></a>
</h3>
<p>Los datos de esta página pueden descargarse con las instrucciones de la página <a href="download-handbook-and-data.html#download-handbook-and-data">Descargando el manual y los datos</a>.</p>
<p>Hay varios formatos diferentes en los que se puede almacenar un árbol filogenético (por ejemplo, Newick, NEXUS, Phylip). Uno de los más comunes es el formato de archivo Newick (.nwk), que es el estándar para representar árboles en forma legible por el ordenador. Esto significa que un árbol completo puede expresarse en un formato de cadena como “((t2:0,04,t1:0,34):0,89,(t5:0,37,(t4:0,03,t3:0,67):0,9):0,59);”, enumerando todos los nodos y puntas, y su relación (longitud de rama) entre sí.</p>
<p>Nota: Es importante entender que el archivo del árbol filogenético en sí mismo no contiene datos de secuenciación, sino que es simplemente el resultado de las distancias genéticas entre las secuencias. Por lo tanto, no podemos extraer datos de secuenciación de un archivo de árbol.</p>
<p>En primer lugar, utilizamos la función read.tree() del paquete <strong>ape</strong> para importar un archivo de árbol filogenético de Newick en formato .txt, y lo almacenamos en un objeto tipo lista llamado “phylo”. Si es necesario, utiliza la función <code><a href="https://here.r-lib.org//reference/here.html">here()</a></code> del paquete <strong>here</strong> para especificar la ruta relativa del archivo.</p>
<p>Nota: En este caso el árbol newick se guarda como un archivo .txt para facilitar su manejo y descarga desde Github.</p>
<div class="sourceCode" id="cb1736"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="st">"Shigella_tree.txt"</span><span class="op">)</span></code></pre></div>
<p>Inspeccionamos nuestro objeto árbol (‘tree’) y vemos que contiene 299 puntas (o muestras) y 236 nodos.</p>
<div class="sourceCode" id="cb1737"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 299 tips and 236 internal nodes.
## 
## Tip labels:
##   SRR5006072, SRR4192106, S18BD07865, S18BD00489, S17BD08906, S17BD05939, ...
## Node labels:
##   17, 29, 100, 67, 100, 100, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>En segundo lugar, importamos una tabla almacenada en un archivo .csv con información adicional para cada muestra secuenciada, como el sexo, el país de origen y los atributos de resistencia antimicrobiana, utilizando la función <code><a href="https://rdrr.io/pkg/rio/man/import.html">import()</a></code> del paquete <strong>rio</strong>:</p>
<div class="sourceCode" id="cb1739"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rio/man/import.html">import</a></span><span class="op">(</span><span class="st">"sample_data_Shigella_tree.csv"</span><span class="op">)</span></code></pre></div>
<p>A continuación se muestran las primeras 50 filas de los datos:</p>
<div id="htmlwidget-fc9d675d5928d41e2ecf" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fc9d675d5928d41e2ecf">{"x":{"filter":"none","vertical":false,"data":[["S17BD05944","S15BD07413","S18BD07247","S19BD07384","S18BD07338","S18BD02657","S19BD06884","S16BD02574","S16BD04069","S17BD08170","S15BD07250","S17BD07935","S17BD00384","S19BD06034","S17BD08948","S17BD04542","S17BD07399","S17BD08233","S19BD07381","S17BD07961","S18BD09176","S19BD07393","S18BD02580","S17BD02526","S17BD06833","S18BD01024","S18BD03411","S18BD09515","S17BD07332","S18BD06598","S17BD02164","S19BD06386","S19BD06283","S19BD06285","S19BD05221","S18BD06749","S17BD02088","S17BD07692","S14BD03573","S17BD01135","S19BD06689","S15BD06969","S18BD01106","S17BD00525","S19BD07036","S18BD06401","S18BD06486","S17BD06426","S17BD05916","S18BD07708"],["Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei"],["Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium"],["Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe"],[null,"Georgia",null,"Morocco",null,null,null,"Thailand",null,null,null,null,null,null,null,"Iran",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Morocco","France","Turkey","Egypt","Tunisia","Congo",null,null,null,null,null,null,null,null,null,null,"Egypt",null,null],["2017","2015","2018","2019","2018","2018","2019","2016","2016","2017","2015","2017","2017","2019","2017","2017","2017","2017","2019","2017","2018","2019","2018","2017","2017","2018","2018","2018","2017","2018","2017","2019","2019","2019","2019","2018","2017","2017","2014","2017","2019","2015","2018","2017","2019","2018","2018","2017","2017","2018"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL"],[null,"Male","Female","Female","Male","Male","Male","Male","Male","Male","Female",null,"Male","Male","Female","Male","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Male",null,"female","Male","Female","Male","Male","Male","Male","Male","Male","Male","Female"],[null,null,null,null,null,"gyrA D87Y","gyrA D87Y","gyrA D87Y","gyrA D87Y",null,null,"gyrA D87Y",null,null,"gyrA D87Y",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA S83L",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA D87Y",null,null,null,"gyrA D87Y",null,null,null,null,null,null,null],[null,null,null,null,null,"mphA","mphA",null,null,null,null,"mphA",null,null,"mphA",null,null,null,null,null,null,null,null,null,null,"mphA","mphA",null,null,null,null,null,null,null,"mphA/ermB",null,null,"mphA",null,null,null,null,"mphA",null,null,null,null,null,null,null],[8,0,16,8,8,64,64,0,0,64,0,64,8,8,64,8,8,8,8,8,8,16,8,8,8,64,64,8,8,8,8,8,8,2,64,8,8,64,0,8,8,0,64,8,4,8,8,8,8,8],[0.25,0,0.03,0.02,0.02,0.5,0.5,0,0,8,0,0.5,0.02,0.02,0.5,0.25,0.02,0.02,0.02,0.02,0.02,0.12,0.02,0.02,0.02,0.5,0.25,0.03,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,1,0,0.02,0.02,0,0.5,0.02,0.02,4,0.02,0.02,0.25,0.02]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>serotype<\/th>\n      <th>Country<\/th>\n      <th>Continent<\/th>\n      <th>Travel_history<\/th>\n      <th>Year<\/th>\n      <th>Belgium<\/th>\n      <th>Source<\/th>\n      <th>Gender<\/th>\n      <th>gyrA_mutations<\/th>\n      <th>macrolide_resistance_genes<\/th>\n      <th>MIC_AZM<\/th>\n      <th>MIC_CIP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[11,12]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="limpiar-e-inspeccionar" class="section level3 unnumbered">
<h3>Limpiar e inspeccionar<a class="anchor" aria-label="anchor" href="#limpiar-e-inspeccionar"><i class="fas fa-link"></i></a>
</h3>
<p>Limpiamos e inspeccionamos nuestros datos: Para asignar los datos de muestra correctos al árbol filogenético, los valores de la columna <code>Sample_ID</code> en el dataframe <code>sample_data</code> deben coincidir con los valores de <code>tip.labels</code> en el archivo <code>tree</code>:</p>
<p>Comprobamos el formato de los <code>tip.labels</code> en el archivo de árbol mirando las 6 primeras entradas usando <code><a href="https://rdrr.io/r/utils/head.html">head()</a></code> de R <strong>base</strong>.</p>
<div class="sourceCode" id="cb1740"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span> </code></pre></div>
<pre><code>## [1] "SRR5006072" "SRR4192106" "S18BD07865" "S18BD00489"
## [5] "S17BD08906" "S17BD05939"</code></pre>
<p>También nos aseguramos de que la primera columna de nuestro dataframe <code>sample_data</code> sea <code>Sample_ID</code>. Miramos los nombres de las columnas de nuestro dataframe utilizando <code><a href="https://rdrr.io/r/base/colnames.html">colnames()</a></code> de R <strong>base</strong>.</p>
<div class="sourceCode" id="cb1742"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span>   </code></pre></div>
<pre><code>##  [1] "Sample_ID"                  "serotype"                  
##  [3] "Country"                    "Continent"                 
##  [5] "Travel_history"             "Year"                      
##  [7] "Belgium"                    "Source"                    
##  [9] "Gender"                     "gyrA_mutations"            
## [11] "macrolide_resistance_genes" "MIC_AZM"                   
## [13] "MIC_CIP"</code></pre>
<p>Miramos los <code>Sample_IDs</code> en el dataframe para asegurarnos de que el formato es el mismo que en el <code>tip.label</code> (por ejemplo, las letras son todas mayúsculas, no hay barras bajas adicionales <code>_</code> entre las letras y los números, etc.)</p>
<div class="sourceCode" id="cb1744"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">)</span> <span class="co"># volvemos a inspeccionar sólo los 6 primeros usando head()</span></code></pre></div>
<pre><code>## [1] "S17BD05944" "S15BD07413" "S18BD07247" "S19BD07384"
## [5] "S18BD07338" "S18BD02657"</code></pre>
<p>También podemos comparar si todas las muestras están presentes en el archivo <code>tree</code> y viceversa, generando un vector lógico de TRUE o FALSE donde coinciden o no. Estos no se imprimen aquí, para simplificar.</p>
<div class="sourceCode" id="cb1746"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span> <span class="op">%in%</span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>

<span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">%in%</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Podemos utilizar estos vectores para mostrar cualquier ID que no esté en el árbol (no hay ninguno).</p>
<div class="sourceCode" id="cb1747"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">[</span><span class="op">!</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">%in%</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">]</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>Al inspeccionar podemos ver que el formato de <code>Sample_ID</code> en el dataframe corresponde al formato de los nombres de las muestras en el <code>tip.labels</code>. No es necesario que estén clasificados en el mismo orden para que coincidan.</p>
<p>Estamos listos para empezar!</p>
<!-- ======================================================= -->
</div>
</div>
<div id="simple-tree-visualization" class="section level2" number="38.3">
<h2>
<span class="header-section-number">38.3</span> Visualización simple de un árbol<a class="anchor" aria-label="anchor" href="#simple-tree-visualization"><i class="fas fa-link"></i></a>
</h2>
<div id="diferentes-configuraciones-de-los-árboles" class="section level3 unnumbered">
<h3>Diferentes configuraciones de los árboles<a class="anchor" aria-label="anchor" href="#diferentes-configuraciones-de-los-%C3%A1rboles"><i class="fas fa-link"></i></a>
</h3>
<p><strong>ggtree</strong> ofrece varios formatos de presentación y algunos pueden ser más adecuados que otros dependiendo del propósito específico. A continuación se muestran algunos ejemplos. Para otras opciones, consulta este <a href="http://yulab-smu.top/treedata-book/chapter4.html">libro en línea</a>.</p>
<p>A continuación, vemos algunos ejemplos de configuración de árboles:</p>
<div class="sourceCode" id="cb1749"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>                                            <span class="co"># árbol lineal simple</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,  branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>                   <span class="co"># árbol lineal simple con todas las puntas alineadas</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span>                         <span class="co"># árbol circular simple</span>
<span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="co"># árbol circular simple con todas las puntas alineadas</span></code></pre></div>
<p><img src="_main_files/figure-html/phylogenetic_trees_example_formats-1.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-2.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-3.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-4.png" width="50%"></p>
</div>
<div id="árbol-simple-con-datos-de-muestra" class="section level3 unnumbered">
<h3>Árbol simple con datos de muestra<a class="anchor" aria-label="anchor" href="#%C3%A1rbol-simple-con-datos-de-muestra"><i class="fas fa-link"></i></a>
</h3>
<p>El operador <strong>%&lt;+%</strong> se utiliza para conectar el dataframe <code>sample_data</code> con el archivo <code>tree</code>. La anotación más sencilla de un árbol es el agregado de los nombres de las muestras en las puntas, así como la coloración de las puntas y, si se desea, de las ramas:</p>
<p>Este es un ejemplo de árbol circular:</p>
<div class="sourceCode" id="cb1750"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout <span class="op">=</span> <span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span> <span class="co"># %&lt;+% agrega un dataframe con datos de muestra al árbol</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">Belgium</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># colorea las ramas de acuerdo con una variable en tu dataframe</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Sample Origin"</span>,                      <span class="co"># nombre de tu esquema de color (aparecerá en la leyenda así) </span>
    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Yes"</span>, <span class="st">"No"</span><span class="op">)</span>,                     <span class="co"># las diferentes opciones en tu variable</span>
    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NRCSS Belgium"</span>, <span class="st">"Other"</span><span class="op">)</span>,        <span class="co"># cómo quieres que se nombren las diferentes opciones en tu leyenda, permite formatearlas</span>
    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span><span class="op">)</span>,                 <span class="co"># el color que deseas asignar a la variable  </span>
    na.value <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span>                        <span class="co"># colorea los valores no disponibles (NA) en negro </span>
  <span class="fu">new_scale_color</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>                             <span class="co"># permite añadir un esquema de color adicional para otra variable</span>
    <span class="fu">geom_tippoint</span><span class="op">(</span>
      mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">Continent</span><span class="op">)</span>,          <span class="co"># color de la punta por continente Puedes cambiar la forma añadiendo "shape = "</span>
      size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span>                               <span class="co"># define el tamaño de la punta</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Continent"</span>,                    <span class="co"># nombre de tu esquema de color (se mostrará en la leyenda así)</span>
    palette <span class="op">=</span> <span class="st">"Set1"</span>,                      <span class="co"># elegimos un conjunto de colores que vienen con el paquete de Brewer</span>
    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span> <span class="op">+</span>                    <span class="co"># para los valores NA elegimos el color gris</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                             <span class="co"># añade el nombre de la muestra en la punta de su rama </span>
    color <span class="op">=</span> <span class="st">'black'</span>,                       <span class="co"># añade tantas líneas de texto como desees con + , pero es posible que tengas que ajustar el valor de desplazamiento para colocarlas una al lado de la otra</span>
    offset <span class="op">=</span> <span class="fl">1</span>,
    size <span class="op">=</span> <span class="fl">1</span>,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span>    
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Phylogenetic tree of Shigella sonnei"</span><span class="op">)</span><span class="op">+</span>       <span class="co"># Nombre del árbol</span>
  <span class="fu">theme</span><span class="op">(</span>
    axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, <span class="co"># elimina el título del eje x</span>
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, <span class="co"># elimina el título del eje y</span>
    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>    <span class="co"># define el tamaño y el formato de la fuente del título de la leyenda</span>
      face <span class="op">=</span> <span class="st">"bold"</span>,
      size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,   
    legend.text<span class="op">=</span><span class="fu">element_text</span><span class="op">(</span>       <span class="co"># define el tamaño de letra y tipografía de la leyenda</span>
      face <span class="op">=</span> <span class="st">"bold"</span>,
      size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>      <span class="co"># define el tamaño de letra y tipografía del título del gráfico</span>
      size <span class="op">=</span> <span class="fl">12</span>,
      face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,  
    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,     <span class="co"># define la posición de la leyenda</span>
    legend.box <span class="op">=</span> <span class="st">"vertical"</span>,        <span class="co"># define la posición de la leyenda</span>
    legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>   </code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_adding_sampledata-1.png" width="672" style="display: block; margin: auto;"></div>
<p>Podes exportar el gráfico de árbol con <code>ggsave()</code> como lo harías con cualquier otro objeto ggplot. Escrito de esta manera, ggsave() guarda la última imagen producida en la ruta de archivo que especifiques. Recordá que podes utilizar <code><a href="https://here.r-lib.org//reference/here.html">here()</a></code> y rutas de archivo relativas para guardar fácilmente en subcarpetas, etc.</p>
<div class="sourceCode" id="cb1751"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggsave</span><span class="op">(</span><span class="st">"example_tree_circular_1.png"</span>, width <span class="op">=</span> <span class="fl">12</span>, height <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></code></pre></div>
<!-- ======================================================= -->
</div>
</div>
<div id="tree-manipulation" class="section level2" number="38.4">
<h2>
<span class="header-section-number">38.4</span> Manipulación de árboles<a class="anchor" aria-label="anchor" href="#tree-manipulation"><i class="fas fa-link"></i></a>
</h2>
<p>A veces puede tener un árbol filogenético muy grande y sólo le interesa una parte del árbol. Por ejemplo, si ha creado un árbol que incluye muestras históricas o internacionales para obtener una gran visión general sobre como pueden encajar tus datos en esos contextos. Pero luego, para ver más de cerca alguna parte tus datos, tendrás que inspeccionar sólo esa parte del árbol.</p>
<p>Dado que el archivo del árbol filogenético es el resultado del análisis de los datos de secuenciación, no podemos manipular el orden de los nodos y las ramas en el propio archivo. Estos ya han sido determinados en análisis anteriores a partir de los datos NGS en bruto. Sin embargo, podemos ampliar partes, ocultar partes e incluso subdividir partes del árbol.</p>
<div id="ampliar-el-zoom" class="section level3 unnumbered">
<h3>Ampliar el zoom<a class="anchor" aria-label="anchor" href="#ampliar-el-zoom"><i class="fas fa-link"></i></a>
</h3>
<p>Si en vez de “cortar” tu árbol, prefieres inspeccionar sólo una parte más de cerca, puedes hacer zoom para ver una parte específica.</p>
<p>En primer lugar, trazamos todo el árbol en formato lineal y añadimos etiquetas numéricas a cada nodo del árbol.</p>
<div class="sourceCode" id="cb1752"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,<span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>                <span class="co"># etiqueta las puntas de todas las ramas del árbol con el nombre de la muestra</span>
  <span class="fu">geom_text2</span><span class="op">(</span>
    mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>,
                  label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,
    size <span class="op">=</span> <span class="fl">5</span>,
    color <span class="op">=</span> <span class="st">"darkred"</span>,
    hjust <span class="op">=</span> <span class="fl">1</span>,
    vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>                            <span class="co"># etiqueta todos los nodos del árbol</span>

<span class="va">p</span>  <span class="co"># imprime en pantalla</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Para hacer zoom en una rama en particular (la que sobresale a la derecha), utilizá <code>viewClade()</code> en el objeto ggtree <code>p</code> y proporcioná el número de nodo para verlo más de cerca:</p>
<div class="sourceCode" id="cb1753"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">viewClade</span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in_452-1.png" width="50%" style="display: block; margin: auto;"></div>
</div>
<div id="ramas-colapsadas" class="section level3 unnumbered">
<h3>Ramas colapsadas<a class="anchor" aria-label="anchor" href="#ramas-colapsadas"><i class="fas fa-link"></i></a>
</h3>
<p>Sin embargo, tal vez queramos ignorar esta rama, entonces podemos colapsarla en ese mismo nodo (nodo 452) utilizando <code>collapse()</code>. Definimos este árbol como <code>p_collapsed</code>.</p>
<div class="sourceCode" id="cb1754"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_collapsed</span> <span class="op">&lt;-</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span>
<span class="va">p_collapsed</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_collapse_452-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Como aclaración, cuando imprimimos p_collapsed, añadimos un <code>geom_point2()</code> (un diamante azul) en el nodo de la rama colapsada.</p>
<div class="sourceCode" id="cb1755"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_collapsed</span> <span class="op">+</span> 
<span class="fu">geom_point2</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">(</span><span class="va">node</span> <span class="op">==</span> <span class="fl">452</span><span class="op">)</span><span class="op">)</span>,  <span class="co"># asignamos un símbolo al nodo colapsado</span>
            size <span class="op">=</span> <span class="fl">5</span>,                     <span class="co"># definimos el tamaño del símbolo</span>
            shape <span class="op">=</span> <span class="fl">23</span>,                   <span class="co"># definimos la forma del símbolo</span>
            fill <span class="op">=</span> <span class="st">"steelblue"</span><span class="op">)</span>           <span class="co"># definimos el color del símbolo</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1319-1.png" width="672"></div>
</div>
<div id="subconjunto-de-un-árbol" class="section level3 unnumbered">
<h3>Subconjunto de un árbol<a class="anchor" aria-label="anchor" href="#subconjunto-de-un-%C3%A1rbol"><i class="fas fa-link"></i></a>
</h3>
<p>Si queremos hacer un cambio más permanente y crear un nuevo árbol reducido con el que trabajar, podemos subconjuntar parte de él con <code>tree_subset()</code>. Luego podemos guardarlo como un nuevo archivo de árbol newick o archivo .txt.</p>
<p>En primer lugar, inspeccionamos los nodos del árbol y las etiquetas de las puntas para decidir qué subconjunto se va a seleccionar.</p>
<div class="sourceCode" id="cb1756"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span>
  <span class="va">tree</span>,
  branch.length <span class="op">=</span> <span class="st">'none'</span>,
  layout <span class="op">=</span> <span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>               <span class="co"># añade los datos de la muestra usando el operador %&lt;+%</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">+</span>                                <span class="co"># etiqueta las puntas de todas las ramas del árbol con el nombre de la muestra</span>
  <span class="fu">geom_text2</span><span class="op">(</span>
    mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>, label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    color <span class="op">=</span> <span class="st">"darkred"</span><span class="op">)</span> <span class="op">+</span>                                <span class="co"># etiqueta todos los nodos del árbol</span>
 <span class="fu">theme</span><span class="op">(</span>
   legend.position <span class="op">=</span> <span class="st">"none"</span>,                            <span class="co"># elimina la leyenda</span>
   axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
   axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
   plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_subsetting-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Ahora, digamos que hemos decidido crear un un subconjunto del árbol (o sub-árbol) con solo el nodo 528 (manteniendo las puntas dentro de esta rama después del nodo 528) y lo guardamos como un nuevo objeto <code>sub_tree1</code>:</p>
<div class="sourceCode" id="cb1757"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sub_tree1</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span>
  <span class="va">tree</span>,
  node <span class="op">=</span> <span class="fl">528</span><span class="op">)</span>                                            <span class="co">#Subconjuntamos el árbol en el nodo 528</span></code></pre></div>
<p>Veamos el subconjunto del árbol 1:</p>
<div class="sourceCode" id="cb1758"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Subset tree 1"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1321-1.png" width="672"></div>
<p>También podes hacer un subconjunto basado en una muestra en particular, especificando cuántos nodos “hacia atrás” queres incluir. Vamos a subconjuntar la misma parte del árbol basándonos en una muestra, en este caso S17BD07692, retrocediendo 9 nodos y lo guardamos como un nuevo objeto <code>sub_tree2</code>:</p>
<div class="sourceCode" id="cb1759"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sub_tree2</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span>
  <span class="va">tree</span>,
  <span class="st">"S17BD07692"</span>,
  levels_back <span class="op">=</span> <span class="fl">9</span><span class="op">)</span> <span class="co"># levels back define cuántos nodos hacia atrás quieres ir desde la punta de la muestra</span></code></pre></div>
<p>Veamos el subconjunto del árbol 2:</p>
<div class="sourceCode" id="cb1760"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span>  <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Subset tree 2"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1323-1.png" width="672"></div>
<p>También podes guardar tu nuevo sub-árbol como un archivo Newick o incluso un archivo de texto utilizando la función <code>write.tree()</code> del paquete <strong>ape</strong>:</p>
<div class="sourceCode" id="cb1761"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># para guardar en formato .nwk </span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.nwk'</span><span class="op">)</span>

<span class="co"># para guardar en formato  .txt</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.txt'</span><span class="op">)</span></code></pre></div>
</div>
<div id="rotación-de-nodos-en-un-árbol" class="section level3 unnumbered">
<h3>Rotación de nodos en un árbol<a class="anchor" aria-label="anchor" href="#rotaci%C3%B3n-de-nodos-en-un-%C3%A1rbol"><i class="fas fa-link"></i></a>
</h3>
<p>Como ya hemos dicho, no podemos cambiar el orden de las puntas o de los nodos en el árbol, ya que éste se basa en su parentesco genético y no está sujeto a manipulación visual. Pero podemos rotar las ramas alrededor de los nodos si eso facilita la visualización.</p>
<p>En primer lugar, trazamos nuestro nuevo sub-árbol 2 con las etiquetas de los nodos para elegir el nodo que queremos manipular y lo almacenamos en un objeto de ggtree plot <code>p</code>.</p>
<div class="sourceCode" id="cb1762"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text2</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="op">!</span><span class="va">isTip</span>, label<span class="op">=</span><span class="va">node</span><span class="op">)</span>, <span class="co"># etiqueta todos los nodos del árbol</span>
             size <span class="op">=</span> <span class="fl">5</span>,
             color <span class="op">=</span> <span class="st">"darkred"</span>, 
             hjust <span class="op">=</span> <span class="fl">1</span>, 
             vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> 
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_1-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Luego podemos manipular los nodos aplicando ggtree<strong>::rotate()</strong> o <strong>ggtree::flip()</strong>: Nota: para ilustrar qué nodos estamos manipulando aplicamos primero la función <strong>geom_hilight()</strong> de <strong>ggtree</strong> para resaltar las muestras en los nodos que nos interesan y almacenamos ese objeto gráfico de ggtree en un nuevo objeto <code>p1</code>.</p>
<div class="sourceCode" id="cb1763"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu">geom_hilight</span><span class="op">(</span>  <span class="co"># resalta el nodo 39 en azul, "extend =" permite definir la longitud del bloque de color</span>
  node <span class="op">=</span> <span class="fl">39</span>,
  fill <span class="op">=</span> <span class="st">"steelblue"</span>,
  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>  
<span class="fu">geom_hilight</span><span class="op">(</span>            <span class="co"># resalta el nodo 37 en amarillo</span>
  node <span class="op">=</span> <span class="fl">37</span>,
  fill <span class="op">=</span> <span class="st">"yellow"</span>,
  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>               
<span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Original tree"</span><span class="op">)</span>


<span class="va">p1</span> <span class="co"># imprime en pantalla</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_2-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Ahora podemos rotar el nodo 37 en el objeto <code>p1</code> para que las muestras del nodo 38 se muevan hacia abajo. Almacenamos el árbol rotado en un nuevo objeto p2</p>
<div class="sourceCode" id="cb1764"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/rotate.html">rotate</a></span><span class="op">(</span><span class="va">p1</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span> <span class="co">#jfmont</span>
      <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Rotated Node 37"</span><span class="op">)</span>


<span class="va">p2</span>   <span class="co"># imprime en pantalla</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1324-1.png" width="672"></div>
<p>O podemos usar el comando flip para rotar el nodo 36 en el objeto <code>p1</code> y mover el nodo 37 a la parte superior y el nodo 39 a la parte inferior. Almacenamos el árbol con nodos rotados como un nuevo objeto <code>p3</code>.</p>
<div class="sourceCode" id="cb1765"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p3</span> <span class="op">&lt;-</span>  <span class="fu">flip</span><span class="op">(</span><span class="va">p1</span>, <span class="fl">39</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Rotated Node 36"</span><span class="op">)</span>


<span class="va">p3</span>   <span class="co"># imprime en pantalla</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1325-1.png" width="672"></div>
</div>
<div id="ejemplo-de-sub-árbol-con-anotación-de-datos" class="section level3 unnumbered">
<h3>Ejemplo de sub-árbol con anotación de datos<a class="anchor" aria-label="anchor" href="#ejemplo-de-sub-%C3%A1rbol-con-anotaci%C3%B3n-de-datos"><i class="fas fa-link"></i></a>
</h3>
<p>Digamos que estamos investigando el grupo de casos con expansión clonal que se produjo en 2017 y 2018 representados en el nodo 39 de nuestro sub-árbol. Añadimos el año de aislamiento de la cepa, así como el historial de viajes y coloreamos por país para ver el origen de otras cepas estrechamente relacionadas genéticamente:</p>
<div class="sourceCode" id="cb1766"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>     <span class="co"># usamos el operador %&lt;+% para enlazar con sample_data</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># etiqueta las puntas de todas las ramas del árbol con el nombre de la muestra</span>
    size <span class="op">=</span> <span class="fl">2.5</span>,
    offset <span class="op">=</span> <span class="fl">0.001</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># establece los límites del eje x de nuestro árbol</span>
  <span class="fu">geom_tippoint</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">Country</span><span class="op">)</span>,     <span class="co"># colorea la punta por continente</span>
                size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">scale_color_brewer</span><span class="op">(</span>
    name <span class="op">=</span> <span class="st">"Country"</span>, 
    palette <span class="op">=</span> <span class="st">"Set1"</span>, 
    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>                         <span class="co"># añade el año de aislamiento como etiqueta de texto en las puntas</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">Year</span><span class="op">)</span>,
    color <span class="op">=</span> <span class="st">'blue'</span>,
    offset <span class="op">=</span> <span class="fl">0.0045</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    linetype <span class="op">=</span> <span class="st">"blank"</span> ,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># añade el historial de viajes como una etiqueta de texto en las puntas, en color rojo</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">Travel_history</span><span class="op">)</span>,
    color <span class="op">=</span> <span class="st">'red'</span>,
    offset <span class="op">=</span> <span class="fl">0.006</span>,
    size <span class="op">=</span> <span class="fl">3</span>,
    linetype <span class="op">=</span> <span class="st">"blank"</span>,
    geom <span class="op">=</span> <span class="st">"text"</span>,
    align <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> 
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Phylogenetic tree of Belgian S. sonnei strains with travel history"</span><span class="op">)</span><span class="op">+</span>  <span class="co"># añade el título del árbol</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"genetic distance (0.001 = 4 nucleotides difference)"</span><span class="op">)</span><span class="op">+</span>                   <span class="co"># añade una etiqueta en el eje x</span>
 
  <span class="fu">theme</span><span class="op">(</span>
    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_inspect_subset_example-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Nuestra observación apunta a un evento de importación de cepas procedentes de Asia, que luego circularon en Bélgica a lo largo de los años y parecen haber causado el último brote.</p>
<!-- ======================================================= -->
</div>
</div>
<div id="more-complex-trees-adding-heatmaps-of-sample-data" class="section level2" number="38.5">
<h2>
<span class="header-section-number">38.5</span> Árboles más complejos: añadir mapas térmicos de datos de muestra<a class="anchor" aria-label="anchor" href="#more-complex-trees-adding-heatmaps-of-sample-data"><i class="fas fa-link"></i></a>
</h2>
<p>Podemos añadir información más compleja, como la presencia categórica de genes de resistencia antimicrobiana y valores numéricos de resistencia realmente medida contra agentes antimicrobianos en forma de mapa de calor utilizando la función <strong>ggtree::gheatmap()</strong>.</p>
<p>Primero necesitamos graficar nuestro árbol (puede ser lineal o circular) y almacenarlo en un nuevo objeto ggtree <code>p</code>: Utilizaremos el sub-árbol de la parte 3).</p>
<div class="sourceCode" id="cb1767"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span>, branch.length<span class="op">=</span><span class="st">'none'</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span> 
 <span class="fu">theme</span><span class="op">(</span>
   legend.position <span class="op">=</span> <span class="st">"none"</span>,
    axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>
      size <span class="op">=</span> <span class="fl">12</span>,
      face <span class="op">=</span> <span class="st">"bold"</span>,
      hjust <span class="op">=</span> <span class="fl">0.5</span>,
      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap-1.png" width="60%" style="display: block; margin: auto;"></div>
<p>En segundo lugar, preparamos nuestros datos. Para visualizar las diferentes variables con nuevos esquemas de color, realizamos un subconjunto de nuestro dataframe a la variable deseada. Es importante añadir el <code>Sample_ID</code> como nombre de fila (rownames) de lo contrario los datos no coinciden con los <code>tip.labels</code> del árbol.</p>
<p>En nuestro ejemplo, queremos ver el género y las mutaciones que podrían conferir resistencia a la ciprofloxacina, un importante antibiótico de primera línea utilizado para tratar las infecciones por Shigella.</p>
<p>Creamos un dataframe para el género:</p>
<div class="sourceCode" id="cb1768"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"gender"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gender"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gender</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un dataframe para las mutaciones en el gen gyrA, que confieren resistencia a la ciprofloxacina:</p>
<div class="sourceCode" id="cb1769"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cipR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"cipR"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gyrA_mutations"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cipR</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un dataframe para la concentración inhibitoria mínima (CIM) medida en laboratorio para la ciprofloxacina:</p>
<div class="sourceCode" id="cb1770"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MIC_Cip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"mic_cip"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MIC_CIP"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">MIC_Cip</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></code></pre></div>
<p>Creamos un primer gráfico añadiendo un mapa de calor binario para el género al árbol filogenético y almacenándolo en un nuevo objeto de gráfico ggtree <code>h1</code>:</p>
<div class="sourceCode" id="cb1771"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>,                            <span class="co"># añadimos al árbol una capa de mapa de calor para el género del dataframe</span>
                offset <span class="op">=</span> <span class="fl">10</span>,                          <span class="co"># offset desplaza el mapa de calor a la derecha</span>
                width <span class="op">=</span> <span class="fl">0.10</span>,                         <span class="co"># width define el ancho de la columna del mapa de calor</span>
                color <span class="op">=</span> <span class="cn">NULL</span>,                         <span class="co"># color define el borde de las columnas del mapa de calor</span>
         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>                          <span class="co"># oculta los nombres de las columnas del mapa de calor</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,                  <span class="co"># define el esquema de colores y la leyenda para el género</span>
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1774"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_gender-1.png" width="70%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos información sobre las mutaciones en el gen gyrA, que confieren resistencia a la ciprofloxacina:</p>
<p>Nota: La presencia de mutaciones cromosómicas puntuales en los datos de secuenciación del genoma completo (WGS) se determinó previamente utilizando la herramienta PointFinder desarrollada por Zankari et al. (ver la sección de referencias adicionales)</p>
<p>En primer lugar, asignamos un nuevo esquema de colores a nuestro objeto gráfico <code>h1</code> y lo almacenamos en un objeto llamado <code>h2</code>. Esto nos permite definir y cambiar los colores para nuestra segunda variable en el mapa de calor.</p>
<div class="sourceCode" id="cb1775"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>A continuación, añadimos la segunda capa del mapa de calor a <code>h2</code> y almacenamos los gráficos combinados en un nuevo objeto <code>h3</code>:</p>
<div class="sourceCode" id="cb1776"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,         <span class="co"># añade la segunda capa del mapa de calor que describe las mutaciones de resistencia a la ciprofloxacina</span>
               offset <span class="op">=</span> <span class="fl">12</span>, 
               width <span class="op">=</span> <span class="fl">0.10</span>, 
               colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Ciprofloxacin resistance \n conferring mutation"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1779"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h3</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_genes-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Repetimos el proceso anterior, añadiendo primero una nueva capa de escala de colores a nuestro objeto existente <code>h3</code>, y luego añadiendo los datos continuos sobre la concentración inhibitoria mínima (CIM) de ciprofloxacina para cada cepa al objeto resultante <code>h4</code> para producir el objeto final <code>h5</code>:</p>
<div class="sourceCode" id="cb1780"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Primero añadimos el nuevo esquema de colores:</span>
<span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>

<span class="co"># luego combinamos los dos en una nueva gráfica:</span>
<span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>,  
               offset <span class="op">=</span> <span class="fl">14</span>, 
               width <span class="op">=</span> <span class="fl">0.10</span>,
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC for Ciprofloxacin"</span>,  <span class="co"># aquí definimos un esquema de color de gradiente para la variable continua MIC</span>
                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,
                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.50</span>, <span class="fl">1.00</span><span class="op">)</span>,
                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colourbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1783"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_MIC-1.png" width="90%" style="display: block; margin: auto;"></div>
<p>Podemos hacer el mismo ejercicio para un árbol lineal:</p>
<div class="sourceCode" id="cb1784"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="co"># etiqueta las puntas</span>
  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"genetic distance (0.001 = 4 nucleotides difference)"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,
      axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
      plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, 
                                face <span class="op">=</span> <span class="st">"bold"</span>,
                                hjust <span class="op">=</span> <span class="fl">0.5</span>,
                                vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_1-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Primero añadimos el género:</p>
<div class="sourceCode" id="cb1785"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>, 
                offset <span class="op">=</span> <span class="fl">0.003</span>,
                width <span class="op">=</span> <span class="fl">0.1</span>, 
                color<span class="op">=</span><span class="st">"black"</span>, 
         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1788"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h1</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_2-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos las mutaciones de resistencia a la ciprofloxacina después de añadir otra capa de colores representando los genes que confieren resistencia antimicrobiana:</p>
<div class="sourceCode" id="cb1789"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>
<span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,   
               offset <span class="op">=</span> <span class="fl">0.004</span>, 
               width <span class="op">=</span> <span class="fl">0.1</span>,
               color <span class="op">=</span> <span class="st">"black"</span>,
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Ciprofloxacin resistance \n conferring mutation"</span>,
                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,
                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,
                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1792"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="va">h3</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_3-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>A continuación, añadimos en forma de mapa de calor la concentración mínima inhibitoria determinada por el laboratorio (MIC):</p>
<div class="sourceCode" id="cb1793"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span>
<span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>, 
               offset <span class="op">=</span> <span class="fl">0.005</span>,  
               width <span class="op">=</span> <span class="fl">0.1</span>,
               color <span class="op">=</span> <span class="st">"black"</span>, 
                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_fill_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC for Ciprofloxacin"</span>,
                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,
                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.50</span>,<span class="fl">1.00</span><span class="op">)</span>,
                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_colourbar</span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
   <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,
        legend.box <span class="op">=</span> <span class="st">"horizontal"</span>, legend.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>shape <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y',
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for 'fill' is already present. Adding another scale for
## 'fill', which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1796"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_4-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="resources-29" class="section level2" number="38.6">
<h2>
<span class="header-section-number">38.6</span> Recursos<a class="anchor" aria-label="anchor" href="#resources-29"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#" class="uri">http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#</a> Clade_Colors</p>
<p><a href="https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a></p>
<p><a href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html" class="uri">https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html</a></p>
<p><a href="https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a></p>
<p>Ea Zankari, Rosa Allesøe, Katrine G Joensen, Lina M Cavaco, Ole Lund, Frank M Aarestrup, PointFinder: una novedosa herramienta web para la detección basada en WGS de la resistencia a los antimicrobianos asociada a mutaciones puntuales cromosómicas en patógenos bacterianos, Journal of Antimicrobial Chemotherapy, Volume 72, Issue 10, October 2017, Pages 2764-2768, <a href="https://doi.org/10.1093/jac/dkx217">https://doi.org/10.1093/jac/dkx217</a></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="transmission-chains.html"><span class="header-section-number">37</span> Cadenas de transmisión</a></div>
<div class="next"><a href="interactive-plots.html"><span class="header-section-number">39</span> Gráficos interactivos</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>En esta página</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#phylogenetic-trees-1"><span class="header-section-number">38</span> Árboles filogenéticos</a></li>
<li><a class="nav-link" href="#overview-8"><span class="header-section-number">38.1</span> Resumen</a></li>
<li>
<a class="nav-link" href="#preparation-31"><span class="header-section-number">38.2</span> Preparación</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#cargar-paquetes-27">Cargar paquetes</a></li>
<li><a class="nav-link" href="#importar-datos-22">Importar datos</a></li>
<li><a class="nav-link" href="#limpiar-e-inspeccionar">Limpiar e inspeccionar</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#simple-tree-visualization"><span class="header-section-number">38.3</span> Visualización simple de un árbol</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#diferentes-configuraciones-de-los-%C3%A1rboles">Diferentes configuraciones de los árboles</a></li>
<li><a class="nav-link" href="#%C3%A1rbol-simple-con-datos-de-muestra">Árbol simple con datos de muestra</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#tree-manipulation"><span class="header-section-number">38.4</span> Manipulación de árboles</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#ampliar-el-zoom">Ampliar el zoom</a></li>
<li><a class="nav-link" href="#ramas-colapsadas">Ramas colapsadas</a></li>
<li><a class="nav-link" href="#subconjunto-de-un-%C3%A1rbol">Subconjunto de un árbol</a></li>
<li><a class="nav-link" href="#rotaci%C3%B3n-de-nodos-en-un-%C3%A1rbol">Rotación de nodos en un árbol</a></li>
<li><a class="nav-link" href="#ejemplo-de-sub-%C3%A1rbol-con-anotaci%C3%B3n-de-datos">Ejemplo de sub-árbol con anotación de datos</a></li>
</ul>
</li>
<li><a class="nav-link" href="#more-complex-trees-adding-heatmaps-of-sample-data"><span class="header-section-number">38.5</span> Árboles más complejos: añadir mapas térmicos de datos de muestra</a></li>
<li><a class="nav-link" href="#resources-29"><span class="header-section-number">38.6</span> Recursos</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>EpiRhandbook en español</strong>" fué escrito por el equipo del manual. Compilado el 2022-11-08.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>Este libro fué construido con el paquete <a class="text-light" href="https://bookdown.org">bookdown</a> de R.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
