<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>18 Borrador. Regresión univariante y multivariable | El manual de R para Epi’s</title>
<meta name="author" content="El equipo del manual">
<meta name="description" content="ADVERTENCIA: Esta traducción es sólo un borrador de la traducción al español. Este documento tal cual está ha sido generado automáticamente con DeepL.com y se han hecho algunas correcciones...">
<meta name="generator" content="bookdown 0.29 with bs4_book()">
<meta property="og:title" content="18 Borrador. Regresión univariante y multivariable | El manual de R para Epi’s">
<meta property="og:type" content="book">
<meta property="og:description" content="ADVERTENCIA: Esta traducción es sólo un borrador de la traducción al español. Este documento tal cual está ha sido generado automáticamente con DeepL.com y se han hecho algunas correcciones...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="18 Borrador. Regresión univariante y multivariable | El manual de R para Epi’s">
<meta name="twitter:description" content="ADVERTENCIA: Esta traducción es sólo un borrador de la traducción al español. Este documento tal cual está ha sido generado automáticamente con DeepL.com y se han hecho algunas correcciones...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.0/transition.js"></script><script src="libs/bs3compat-0.4.0/tabs.js"></script><script src="libs/bs3compat-0.4.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.25/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/tabwid-1.1.0/tabwid.css" rel="stylesheet">
<link href="libs/tabwid-1.1.0/scrool.css" rel="stylesheet">
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><script src="libs/d3-4.5.0/d3.min.js"></script><script src="libs/sankey-1/sankey.js"></script><script src="libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><script src="libs/plotly-binding-4.10.0/plotly.js"></script><script src="libs/typedarray-0.1/typedarray.min.js"></script><link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script><link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="libs/vis-9.1.0/vis-network.min.js"></script><script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script><link href="libs/font-awesome-4.7.0/css/font-awesome.min.css" rel="stylesheet">
<script src="libs/plotly-basic-2.5.1/plotly-basic-2.5.1.min.js"></script><script src="libs/plotly-cartesian-2.5.1/plotly-cartesian-2.5.1.min.js"></script><script src="libs/proj4js-2.3.15/proj4.js"></script><link href="libs/highcharts-9.3.1/css/motion.css" rel="stylesheet">
<script src="libs/highcharts-9.3.1/highcharts.js"></script><script src="libs/highcharts-9.3.1/highcharts-3d.js"></script><script src="libs/highcharts-9.3.1/highcharts-more.js"></script><script src="libs/highcharts-9.3.1/modules/stock.js"></script><script src="libs/highcharts-9.3.1/modules/map.js"></script><script src="libs/highcharts-9.3.1/modules/data.js"></script><script src="libs/highcharts-9.3.1/modules/exporting.js"></script><script src="libs/highcharts-9.3.1/modules/offline-exporting.js"></script><script src="libs/highcharts-9.3.1/modules/drilldown.js"></script><script src="libs/highcharts-9.3.1/modules/item-series.js"></script><script src="libs/highcharts-9.3.1/modules/overlapping-datalabels.js"></script><script src="libs/highcharts-9.3.1/modules/annotations.js"></script><script src="libs/highcharts-9.3.1/modules/export-data.js"></script><script src="libs/highcharts-9.3.1/modules/funnel.js"></script><script src="libs/highcharts-9.3.1/modules/heatmap.js"></script><script src="libs/highcharts-9.3.1/modules/treemap.js"></script><script src="libs/highcharts-9.3.1/modules/sankey.js"></script><script src="libs/highcharts-9.3.1/modules/dependency-wheel.js"></script><script src="libs/highcharts-9.3.1/modules/organization.js"></script><script src="libs/highcharts-9.3.1/modules/solid-gauge.js"></script><script src="libs/highcharts-9.3.1/modules/streamgraph.js"></script><script src="libs/highcharts-9.3.1/modules/sunburst.js"></script><script src="libs/highcharts-9.3.1/modules/vector.js"></script><script src="libs/highcharts-9.3.1/modules/wordcloud.js"></script><script src="libs/highcharts-9.3.1/modules/xrange.js"></script><script src="libs/highcharts-9.3.1/modules/tilemap.js"></script><script src="libs/highcharts-9.3.1/modules/venn.js"></script><script src="libs/highcharts-9.3.1/modules/gantt.js"></script><script src="libs/highcharts-9.3.1/modules/timeline.js"></script><script src="libs/highcharts-9.3.1/modules/parallel-coordinates.js"></script><script src="libs/highcharts-9.3.1/modules/bullet.js"></script><script src="libs/highcharts-9.3.1/modules/coloraxis.js"></script><script src="libs/highcharts-9.3.1/modules/dumbbell.js"></script><script src="libs/highcharts-9.3.1/modules/lollipop.js"></script><script src="libs/highcharts-9.3.1/modules/series-label.js"></script><script src="libs/highcharts-9.3.1/plugins/motion.js"></script><script src="libs/highcharts-9.3.1/custom/reset.js"></script><script src="libs/highcharts-9.3.1/modules/boost.js"></script><script src="libs/highchart-binding-0.9.4/highchart.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style_bs4.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">El manual de R para Epi’s</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"></a></li>
<li class="book-part">Acerca de este libro</li>
<li><a class="" href="editorial-and-technical-notes.html"><span class="header-section-number">1</span> Notas editoriales y técnicas</a></li>
<li><a class="" href="download-handbook-and-data.html"><span class="header-section-number">2</span> Descargando el manual y los datos</a></li>
<li class="book-part">Aspectos básicos</li>
<li><a class="" href="r-basics.html"><span class="header-section-number">3</span> Fundamentos de R</a></li>
<li><a class="" href="transition-to-r.html"><span class="header-section-number">4</span> Transición a R</a></li>
<li><a class="" href="suggested-packages-1.html"><span class="header-section-number">5</span> Paquetes recomendados</a></li>
<li><a class="" href="r-projects.html"><span class="header-section-number">6</span> Proyectos en R</a></li>
<li><a class="" href="import-and-export.html"><span class="header-section-number">7</span> Importación y exportación</a></li>
<li class="book-part">Gestión de datos</li>
<li><a class="" href="cleaning-data-and-core-functions.html"><span class="header-section-number">8</span> Limpieza de datos y funciones básicas</a></li>
<li><a class="" href="working-with-dates.html"><span class="header-section-number">9</span> Borrador. Trabajando con Fechas</a></li>
<li><a class="" href="characters-and-strings.html"><span class="header-section-number">10</span> Caracteres y cadenas</a></li>
<li><a class="" href="pivoting-data.html"><span class="header-section-number">11</span> Pivotar datos</a></li>
<li><a class="" href="grouping-data.html"><span class="header-section-number">12</span> Agrupar datos</a></li>
<li><a class="" href="joining-data.html"><span class="header-section-number">13</span> Unir datos</a></li>
<li><a class="" href="de-duplication.html"><span class="header-section-number">14</span> Borrador. De-duplicación</a></li>
<li><a class="" href="iteration-loops-and-lists.html"><span class="header-section-number">15</span> Borrador. Iteración, bucles y listas</a></li>
<li class="book-part">Análisis</li>
<li><a class="" href="descriptive-tables.html"><span class="header-section-number">16</span> Tablas descriptivas</a></li>
<li><a class="" href="simple-statistical-tests.html"><span class="header-section-number">17</span> Tests estadísticos sencillos</a></li>
<li><a class="active" href="univariate-and-multivariable-regression.html"><span class="header-section-number">18</span> Borrador. Regresión univariante y multivariable</a></li>
<li><a class="" href="missing-data.html"><span class="header-section-number">19</span> Borrador. Valores faltantes</a></li>
<li><a class="" href="standardised-rates.html"><span class="header-section-number">20</span> Borrador. Tasas estandarizadas</a></li>
<li><a class="" href="moving-averages.html"><span class="header-section-number">21</span> Borrador. Medias Móviles</a></li>
<li><a class="" href="time-series-and-outbreak-detection.html"><span class="header-section-number">22</span> Borrador. Series temporales y detección de brotes</a></li>
<li><a class="" href="epidemic-modeling.html"><span class="header-section-number">23</span> Borrador. Modelización de epidemias</a></li>
<li><a class="" href="contact-tracing-1.html"><span class="header-section-number">24</span> Borrador. Rastreo de contactos</a></li>
<li><a class="" href="survey-analysis.html"><span class="header-section-number">25</span> Borrador. Análisis de encuestas</a></li>
<li><a class="" href="survival-analysis.html"><span class="header-section-number">26</span> Borrador. Análisis de supervivencia</a></li>
<li><a class="" href="gis-basics.html"><span class="header-section-number">27</span> Conceptos básicos de los SIG</a></li>
<li class="book-part">Visualización de datos</li>
<li><a class="" href="ggplot-basics.html"><span class="header-section-number">28</span> Conceptos básicos de ggplot</a></li>
<li><a class="" href="ggplot-tips.html"><span class="header-section-number">29</span> Consejos de ggplot</a></li>
<li><a class="" href="epidemic-curves.html"><span class="header-section-number">30</span> Curvas epidémicas</a></li>
<li><a class="" href="demographic-pyramids-and-likert-scales.html"><span class="header-section-number">31</span> Pirámides de población y escalas de Likert</a></li>
<li><a class="" href="heat-plots.html"><span class="header-section-number">32</span> Gráficos de calor</a></li>
<li><a class="" href="diagrams-and-charts.html"><span class="header-section-number">33</span> Diagramas y gráficos</a></li>
<li><a class="" href="combinations-analysis.html"><span class="header-section-number">34</span> Borrador. Análisis de combinaciones</a></li>
<li><a class="" href="transmission-chains.html"><span class="header-section-number">35</span> Borrador. Cadenas de transmisión</a></li>
<li><a class="" href="phylogenetic-trees-1.html"><span class="header-section-number">36</span> Árboles filogenéticos</a></li>
<li><a class="" href="interactive-plots.html"><span class="header-section-number">37</span> Gráficos interactivos</a></li>
<li class="book-part">Informes y Dashboards</li>
<li><a class="" href="reports-with-r-markdown.html"><span class="header-section-number">38</span> Informes con R Markdown</a></li>
<li><a class="" href="organizing-routine-reports.html"><span class="header-section-number">39</span> Organización de informes rutinarios</a></li>
<li><a class="" href="dashboards-with-r-markdown.html"><span class="header-section-number">40</span> Borrador. Dashboards con R Markdown</a></li>
<li><a class="" href="dashboards-with-shiny.html"><span class="header-section-number">41</span> Borrador. Dashboards con Shiny</a></li>
<li class="book-part">Miscelánea</li>
<li><a class="" href="writing-functions-1.html"><span class="header-section-number">42</span> Escribir funciones</a></li>
<li><a class="" href="directory-interactions.html"><span class="header-section-number">43</span> Borrador. Interacciones con directorios</a></li>
<li><a class="" href="version-control-and-collaboration-with-git-and-github.html"><span class="header-section-number">44</span> Control de versiones y colaboración con Git y Github</a></li>
<li><a class="" href="common-errors.html"><span class="header-section-number">45</span> Errores comunes</a></li>
<li><a class="" href="getting-help.html"><span class="header-section-number">46</span> Cómo obtener ayuda</a></li>
<li><a class="" href="r-on-network-drives.html"><span class="header-section-number">47</span> Borrador. R en redes locales</a></li>
<li><a class="" href="data-table.html"><span class="header-section-number">48</span> Borrador. data.table</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="univariate-and-multivariable-regression" class="section level1" number="18">
<h1>
<span class="header-section-number">18</span> Borrador. Regresión univariante y multivariable<a class="anchor" aria-label="anchor" href="#univariate-and-multivariable-regression"><i class="fas fa-link"></i></a>
</h1>
<p><span style="color: red;"><strong><em>ADVERTENCIA:</em></strong> Esta traducción es sólo un borrador de la traducción al español. Este documento tal cual está ha sido generado automáticamente con DeepL.com y se han hecho algunas correcciones globales. Está pendiente de una revisión completa. </span></p>
<!-- ======================================================= -->
<p>Esta página muestra el uso de las funciones de regresión <strong>básicas</strong> de R, como <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> y el paquete <strong>gtsummary,</strong> para observar las asociaciones entre variables (por ejemplo, odds ratios, risk ratios y hazard ratios). También utiliza funciones como <code>tidy()</code> del paquete <strong>broom</strong> para limpiar los resultados de la regresión.</p>
<ol style="list-style-type: decimal">
<li>Univariante: tablas de dos en dos</li>
<li>Estratificado: estimaciones mantel-haenszel</li>
<li>Multivariable: selección de variables, selección de modelos, tabla final</li>
<li>Forest plots</li>
</ol>
<p>Para la regresión de riesgos proporcionales de Cox, véase la página de <a href="survival-analysis.html#survival-analysis">análisis de supervivencia</a>.</p>
<p><strong><em>NOTA:</em></strong> Utilizamos el término <em>multivariable</em> para referirnos a una regresión con múltiples variables explicativas. En este sentido, un modelo multivariante sería una regresión con varios resultados - véase esta <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3518362/">editorial</a> para más detalles</p>
<!-- ======================================================= -->
<div id="preparation-9" class="section level2" number="18.1">
<h2>
<span class="header-section-number">18.1</span> Preparación<a class="anchor" aria-label="anchor" href="#preparation-9"><i class="fas fa-link"></i></a>
</h2>
<div id="cargar-paquetes-11" class="section level3 unnumbered">
<h3>Cargar paquetes<a class="anchor" aria-label="anchor" href="#cargar-paquetes-11"><i class="fas fa-link"></i></a>
</h3>
<p>Este trozo de código muestra la carga de los paquetes necesarios para los análisis. En este manual destacamos <code><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load()</a></code> de <strong>pacman</strong>, que instala el paquete si es necesario <em>y</em> lo carga para tu uso. También puede cargar los paquetes instalados con <code><a href="https://rdrr.io/r/base/library.html">library()</a></code> de R <strong>base</strong>. Consulta la página sobre <a href="r-basics.html#r-basics">fundamentos de R</a> para obtener más información sobre los paquetes de R.</p>
<div class="sourceCode" id="cb965"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span>
  <span class="va">rio</span>,          <span class="co"># File import</span>
  <span class="va">here</span>,         <span class="co"># File locator</span>
  <span class="va">tidyverse</span>,    <span class="co"># data management + ggplot2 graphics, </span>
  <span class="va">stringr</span>,      <span class="co"># manipulate text strings </span>
  <span class="va">purrr</span>,        <span class="co"># loop over objects in a tidy way</span>
  <span class="va">gtsummary</span>,    <span class="co"># summary statistics and tests </span>
  <span class="va">broom</span>,        <span class="co"># tidy up results from regressions</span>
  <span class="va">lmtest</span>,       <span class="co"># likelihood-ratio tests</span>
  <span class="va">parameters</span>,   <span class="co"># alternative to tidy up results from regressions</span>
  <span class="va">see</span>          <span class="co"># alternative to visualise forest plots</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="importar-datos-10" class="section level3 unnumbered">
<h3>Importar datos<a class="anchor" aria-label="anchor" href="#importar-datos-10"><i class="fas fa-link"></i></a>
</h3>
<p>Importamos los datos de casos de una epidemia de ébola simulada. Si quiere seguir el proceso, <a href="https://github.com/epirhandbook/Epi_R_handbook/raw/master/data/case_linelists/linelist_cleaned.rds">clica para descargar <code>linelist</code> “limpia”</a> (como archivo .rds). Importa tus datos con la función <code><a href="https://rdrr.io/pkg/rio/man/import.html">import()</a></code> del paquete <strong>rio</strong> (acepta muchos tipos de archivos como .xlsx, .rds, .csv - Mira la página de <a href="import-and-export.html#import-and-export">importación y exportación</a> para más detalles).</p>
<div class="sourceCode" id="cb966"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># import the linelist</span>
<span class="va">linelist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rio/man/import.html">import</a></span><span class="op">(</span><span class="st">"linelist_cleaned.rds"</span><span class="op">)</span></code></pre></div>
<p>A continuación se muestran las primeras 50 filas del listado.</p>
<div id="htmlwidget-9772bc31058d0fea4b47" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9772bc31058d0fea4b47">{"x":{"filter":"none","vertical":false,"data":[["5fe599","8689b7","11f8ea","b8812a","893f25","be99c8","07e3e8","369449","f393b4","1389ca","2978ac","57a565","fc15ef","2eaa9a","bbfa93","c97dd9","f50e8a","3a7673","7f5a01","ddddee","99e8fa","567136","9371a9","bc2adf","403057","8bd1e8","f327be","42e1a9","90e5fe","959170","8ebf6e","e56412","6d788e","a47529","67be4e","da8ecb","148f18","2cb9a5","f5c142","70a9fe","3ad520","062638","c76676","baacc1","497372","23e499","38cc4a","3789ee","c71dcd","6b70f0"],[4,4,2,3,3,3,4,4,4,4,4,4,6,5,6,9,10,8,7,6,7,6,8,6,10,8,6,12,5,8,7,9,11,5,8,5,6,11,7,9,7,8,9,12,13,9,8,10,8,7],["2014-05-08",null,null,"2014-05-04","2014-05-18","2014-05-03","2014-05-22","2014-05-28",null,null,"2014-05-30","2014-05-28","2014-06-14","2014-06-07","2014-06-09",null,null,null,"2014-06-23","2014-06-18","2014-06-24",null,null,"2014-07-03",null,"2014-07-10","2014-06-14",null,"2014-06-18","2014-06-29","2014-07-02","2014-07-12","2014-07-12","2014-06-13","2014-07-15","2014-06-20",null,null,"2014-07-20",null,"2014-07-12","2014-07-19","2014-07-18","2014-07-18","2014-07-27",null,"2014-07-19","2014-07-26","2014-07-24",null],["2014-05-13","2014-05-13","2014-05-16","2014-05-18","2014-05-21","2014-05-22","2014-05-27","2014-06-02","2014-06-05","2014-06-05","2014-06-06","2014-06-13","2014-06-16","2014-06-17","2014-06-18","2014-06-19","2014-06-22","2014-06-23","2014-06-25","2014-06-26","2014-06-28","2014-07-02","2014-07-08","2014-07-09","2014-07-09","2014-07-10","2014-07-12","2014-07-12","2014-07-13","2014-07-13","2014-07-14","2014-07-15","2014-07-16","2014-07-17","2014-07-17","2014-07-18","2014-07-19","2014-07-22","2014-07-22","2014-07-24","2014-07-24","2014-07-25","2014-07-25","2014-07-27","2014-07-29","2014-07-30",null,"2014-08-01","2014-08-02","2014-08-03"],["2014-05-15","2014-05-14","2014-05-18","2014-05-20","2014-05-22","2014-05-23","2014-05-29","2014-06-03","2014-06-06","2014-06-07","2014-06-08","2014-06-15","2014-06-17","2014-06-17","2014-06-20","2014-06-19","2014-06-23","2014-06-24","2014-06-27","2014-06-28","2014-06-29","2014-07-03","2014-07-09","2014-07-09","2014-07-11","2014-07-11","2014-07-13","2014-07-14","2014-07-14","2014-07-13","2014-07-14","2014-07-17","2014-07-17","2014-07-18","2014-07-19","2014-07-20","2014-07-20","2014-07-22","2014-07-24","2014-07-26","2014-07-24","2014-07-27","2014-07-25","2014-07-27","2014-07-31","2014-08-01","2014-08-03","2014-08-02","2014-08-02","2014-08-04"],[null,"2014-05-18","2014-05-30",null,"2014-05-29","2014-05-24","2014-06-01","2014-06-07","2014-06-18","2014-06-09","2014-06-15",null,"2014-07-09",null,"2014-06-30","2014-07-11","2014-07-01","2014-06-25","2014-07-06","2014-07-02","2014-07-09","2014-07-07","2014-07-20",null,"2014-07-22","2014-07-16","2014-07-14","2014-07-20","2014-07-16","2014-07-19","2014-07-27","2014-07-19",null,"2014-07-26","2014-08-14","2014-08-01","2014-07-23","2014-08-28","2014-07-28","2014-07-19",null,"2014-08-03",null,null,null,"2014-08-06","2014-08-21","2014-09-13","2014-08-04",null],[null,"Recover","Recover",null,"Recover","Recover","Recover","Death","Recover","Death","Death","Death","Recover","Recover",null,"Recover",null,null,"Death","Death","Recover",null,null,null,"Death",null,"Death","Death",null,"Death","Recover","Death","Recover","Death","Recover",null,"Death","Recover","Recover","Death",null,null,"Death","Death","Death","Death","Recover",null,"Death","Death"],["m","f","m","f","m","f","f","f","m","f","m","m","m","f","f","m","f","f","f","f","m","m","f","m","f","m","m","f","m","f","f","f","m","m","f","m","f","f","f","m","f","m","f","m","m","f","m","f","m","m"],[2,3,56,18,3,16,16,0,61,27,12,42,19,7,7,13,35,17,11,11,19,54,14,28,6,3,31,6,67,14,10,21,20,45,1,12,3,15,20,36,7,13,14,3,10,1,0,20,26,14],["years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years","years"],[2,3,56,18,3,16,16,0,61,27,12,42,19,7,7,13,35,17,11,11,19,54,14,28,6,3,31,6,67,14,10,21,20,45,1,12,3,15,20,36,7,13,14,3,10,1,0,20,26,14],["0-4","0-4","50-69","15-19","0-4","15-19","15-19","0-4","50-69","20-29","10-14","30-49","15-19","5-9","5-9","10-14","30-49","15-19","10-14","10-14","15-19","50-69","10-14","20-29","5-9","0-4","30-49","5-9","50-69","10-14","10-14","20-29","20-29","30-49","0-4","10-14","0-4","15-19","20-29","30-49","5-9","10-14","10-14","0-4","10-14","0-4","0-4","20-29","20-29","10-14"],["0-4","0-4","55-59","15-19","0-4","15-19","15-19","0-4","60-64","25-29","10-14","40-44","15-19","5-9","5-9","10-14","35-39","15-19","10-14","10-14","15-19","50-54","10-14","25-29","5-9","0-4","30-34","5-9","65-69","10-14","10-14","20-24","20-24","45-49","0-4","10-14","0-4","15-19","20-24","35-39","5-9","10-14","10-14","0-4","10-14","0-4","0-4","20-24","25-29","10-14"],["Other","Missing","St. Mark's Maternity Hospital (SMMH)","Port Hospital","Military Hospital","Port Hospital","Missing","Missing","Missing","Missing","Port Hospital","Military Hospital","Missing","Missing","Other","Port Hospital","Port Hospital","Port Hospital","Missing","Other","Port Hospital","Port Hospital","St. Mark's Maternity Hospital (SMMH)","Missing","Other","Missing","St. Mark's Maternity Hospital (SMMH)","Military Hospital","Port Hospital","Central Hospital","Military Hospital","Central Hospital","Missing","Military Hospital","Other","Missing","Missing","Port Hospital","Port Hospital","Port Hospital","Missing","Central Hospital","Military Hospital","Other","Other","Other","Missing","St. Mark's Maternity Hospital (SMMH)","St. Mark's Maternity Hospital (SMMH)","Missing"],[-13.2157351064963,-13.2152339775486,-13.212910703914,-13.2363711169728,-13.2228638912441,-13.222625321098,-13.2331547837254,-13.2320975453153,-13.2225511595637,-13.2572163655863,-13.2206286746001,-13.253989309478,-13.2385127873491,-13.209391925612,-13.2157278814899,-13.2243437095992,-13.2336087079551,-13.21422143145,-13.2339681355349,-13.2535640411465,-13.2250089377786,-13.2160657166043,-13.2680671272333,-13.2266742923612,-13.2160179088168,-13.2482584611565,-13.2156319199566,-13.2142410663192,-13.2614879104088,-13.2452992638476,-13.2630592726116,-13.2343341712241,-13.2199077448676,-13.2227293309912,-13.2343062806506,-13.218781651651,-13.2483677722899,-13.2097478342339,-13.2680867723786,-13.2587535457526,-13.262635786914,-13.2697246824573,-13.2209026809759,-13.2330734719715,-13.2680923666905,-13.2547212675054,-13.2573683214693,-13.2137356012883,-13.2175973322257,-13.2486407324245],[8.46897295100924,8.45171855856465,8.46481700596819,8.4754761613651,8.46082377490923,8.46183062600728,8.46272931462646,8.46144367534271,8.46191259217774,8.47292327643506,8.48401630165138,8.45837125340844,8.47761705512509,8.47570184950483,8.47779946878972,8.47145134147474,8.47804840685363,8.48528034195779,8.46957530395867,8.45957352078114,8.47404889511544,8.48802917129884,8.473437335922,8.48408263734462,8.46242233645879,8.47026822126572,8.46398447480533,8.4641347894342,8.45623094629607,8.48334624336805,8.47493999153642,8.47832062438022,8.4693933891765,8.48480589906514,8.47121232619015,8.48438437371817,8.48466158574339,8.47714159984428,8.46238127010609,8.45568597813113,8.4632880274758,8.47940722413856,8.46353857052336,8.46178968158864,8.47508713872833,8.45825808128071,8.4532568143863,8.4732571907655,8.47911586641933,8.48480340615605],["f547d6",null,null,"f90f5f","11f8ea","aec8ec","893f25","133ee7",null,null,"996f3a","133ee7","37a6f6","9f6884","4802b1",null,null,null,"a75c7f","8e104d","ab634e",null,null,"b799eb",null,"5d9e4d","a15e13",null,"ea3740","beb26e","567136","894024","36e2e7","a2086d","7baf73","eb2277",null,null,"d6584f",null,"312ecf","52ea64","cfd79c","d145b7","174288",null,"53608c","3b096b","f5c142",null],["other",null,null,"other","other","other","other","other",null,null,"other","other","other","other","other",null,null,null,"other","other","other",null,null,"other",null,"other","other",null,"other","funeral","other","funeral","other","other","other","funeral",null,null,"other",null,"other","other","other","other","other",null,"funeral","other","other",null],[27,25,91,41,36,56,47,0,86,69,67,84,68,44,34,66,78,47,53,47,71,86,53,69,38,46,68,37,100,56,50,57,65,72,29,69,37,48,54,71,47,61,47,35,53,16,13,59,69,67],[48,59,238,135,71,116,87,11,226,174,112,186,174,90,91,152,214,137,117,131,150,241,131,161,80,69,188,66,233,142,110,182,164,214,26,157,39,154,133,168,100,125,123,67,134,31,36,125,183,169],[22,22,21,23,23,21,21,22,22,22,22,22,22,21,23,22,23,21,22,23,21,23,21,24,23,22,24,23,20,24,24,20,24,21,22,21,23,22,23,23,23,22,23,22,22,22,23,22,22,22],["no",null,null,"no","no","no",null,"no","no","no","no","no","no","no","no","no","no","no",null,"no","no","no","no","no",null,"no","no","no",null,null,"no","no",null,"no","no",null,null,"no","no",null,"no","no",null,"no","no","no","no","no","no",null],["no",null,null,"no","no","no",null,"no","no","no","no","no","no","no","no","no","yes","no",null,"no","no","no","yes","no",null,"no","no","yes",null,null,"no","no",null,"no","no",null,null,"no","no",null,"no","no",null,"no","yes","no","no","no","no",null],["yes",null,null,"no","yes","yes",null,"yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes",null,"yes","yes","yes","yes","yes",null,"yes","yes","yes",null,null,"yes","yes",null,"yes","yes",null,null,"yes","yes",null,"yes","yes",null,"yes","yes","yes","yes","yes","no",null],["no",null,null,"no","no","no",null,"no","no","no","no","no","no","no","no","yes","no","no",null,"no","no","no","no","no",null,"no","no","no",null,null,"no","no",null,"no","no",null,null,"yes","yes",null,"no","no",null,"no","no","no","no","no","no",null],["yes",null,null,"no","yes","yes",null,"yes","yes","no","yes","no","no","no","yes","no","no","no",null,"no","yes","no","no","no",null,"no","no","no",null,null,"no","yes",null,"yes","yes",null,null,"yes","yes",null,"yes","yes",null,"yes","yes","no","yes","yes","yes",null],[36.8,36.9,36.9,36.8,36.9,37.6,37.3,37,36.4,35.9,36.5,36.9,36.5,37.1,36.5,37.3,37,38,38,36,37,36.7,36.9,36.5,37,36.5,37.6,36.6,36.6,36.2,36.4,37.1,37.5,37.5,37.4,36.9,36.4,37.3,37,37.8,36.5,37.5,36.7,37,37.3,36.6,36.5,36.6,37.6,36.8],[null,"09:36","16:48","11:22","12:60","14:13","14:33","09:25","11:16","10:55","16:03","11:14","12:42","11:06","09:10","08:45",null,"15:41","13:34","18:58","12:43","16:33","14:29","07:18","08:11","16:32","16:17","07:32","17:45",null,"13:24","14:43","02:33","11:36","17:28","16:27",null,"20:49",null,"11:38","14:25","13:42","21:22","13:33","19:06","17:14","20:09",null,"10:23","09:09"],[117.1875,71.8184429761563,16.0652496292635,22.4965706447188,71.4144019043841,41.6171224732461,62.0953890870657,0,16.8376536925366,22.7903289734443,53.4119897959184,24.2802636142907,22.4600343506408,54.320987654321,41.0578432556455,28.5664819944598,17.0320552013276,25.0412914912888,38.7172182043977,27.3876813705495,31.5555555555556,14.8069075945662,30.8839811199813,26.6193433895297,59.375,96.6183574879227,19.2394748755093,84.9403122130395,18.4199377406104,27.7722674072605,41.3223140495868,17.2080666586161,24.1671624033314,15.7218971089178,428.994082840237,27.9930220292912,243.261012491782,20.2395007589813,30.527446435638,25.15589569161,47,39.04,31.0661643201798,77.9683671196257,29.5165961238583,166.493236212279,100.308641975309,37.76,20.6037803457852,23.458562375267],[2,1,2,2,1,1,2,1,1,2,2,2,1,0,2,0,1,1,2,2,1,1,1,0,2,1,1,2,1,0,0,2,1,1,2,2,1,0,2,2,0,2,0,0,2,2,null,1,0,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>case_id<\/th>\n      <th>generation<\/th>\n      <th>date_infection<\/th>\n      <th>date_onset<\/th>\n      <th>date_hospitalisation<\/th>\n      <th>date_outcome<\/th>\n      <th>outcome<\/th>\n      <th>gender<\/th>\n      <th>age<\/th>\n      <th>age_unit<\/th>\n      <th>age_years<\/th>\n      <th>age_cat<\/th>\n      <th>age_cat5<\/th>\n      <th>hospital<\/th>\n      <th>lon<\/th>\n      <th>lat<\/th>\n      <th>infector<\/th>\n      <th>source<\/th>\n      <th>wt_kg<\/th>\n      <th>ht_cm<\/th>\n      <th>ct_blood<\/th>\n      <th>fever<\/th>\n      <th>chills<\/th>\n      <th>cough<\/th>\n      <th>aches<\/th>\n      <th>vomit<\/th>\n      <th>temp<\/th>\n      <th>time_admission<\/th>\n      <th>bmi<\/th>\n      <th>days_onset_hosp<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,8,10,14,15,18,19,20,26,28,29]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="datos-limpios" class="section level3 unnumbered">
<h3>Datos limpios<a class="anchor" aria-label="anchor" href="#datos-limpios"><i class="fas fa-link"></i></a>
</h3>
<div id="almacenar-las-variables-explicativas" class="section level4 unnumbered">
<h4>Almacenar las variables explicativas<a class="anchor" aria-label="anchor" href="#almacenar-las-variables-explicativas"><i class="fas fa-link"></i></a>
</h4>
<p>Almacenamos los nombres de las columnas explicativas como un vector de caracteres. Esto se referenciará más adelante.</p>
<div class="sourceCode" id="cb967"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## define variables of interest </span>
<span class="va">explanatory_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gender"</span>, <span class="st">"fever"</span>, <span class="st">"chills"</span>, <span class="st">"cough"</span>, <span class="st">"aches"</span>, <span class="st">"vomit"</span><span class="op">)</span></code></pre></div>
</div>
<div id="convertir-a-1s-y-0s" class="section level4 unnumbered">
<h4>Convertir a 1’s y 0’s<a class="anchor" aria-label="anchor" href="#convertir-a-1s-y-0s"><i class="fas fa-link"></i></a>
</h4>
<p>A continuación convertimos las columnas explicativas de “sí”/“no”, “m”/“f”, y “muerto”/“vivo” a <strong>1 / 0</strong>, para cooperar con las expectativas de los modelos de regresión logística. Para hacer esto de manera eficiente, utilizamos <code>across()</code> de <strong>dplyr</strong> para transformar varias columnas a la vez. La función que aplicamos a cada columna es <code>case_when()</code> (también de <strong>dplyr</strong>) que aplica la lógica para convertir los valores especificados en 1’s y 0’s. Mira las secciones sobre <code>across()</code> y <code>case_when()</code> en la página de <a href="cleaning-data-and-core-functions.html#clean_across">Limpieza de datos y funciones básicas</a>).</p>
<p>Nota: el “.” que aparece a continuación representa la columna que está siendo procesada por <code>across()</code> en ese momento.</p>
<div class="sourceCode" id="cb968"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## convert dichotomous variables to 0/1 </span>
<span class="va">linelist</span> <span class="op">&lt;-</span> <span class="va">linelist</span> <span class="op">%&gt;%</span>  
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span>                                      
    .cols <span class="op">=</span> <span class="fu">all_of</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">explanatory_vars</span>, <span class="st">"outcome"</span><span class="op">)</span><span class="op">)</span>,  <span class="co">## for each column listed and "outcome"</span>
    .fns <span class="op">=</span> <span class="op">~</span><span class="fu">case_when</span><span class="op">(</span>                              
      <span class="va">.</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"m"</span>, <span class="st">"yes"</span>, <span class="st">"Death"</span><span class="op">)</span>   <span class="op">~</span> <span class="fl">1</span>,           <span class="co">## recode male, yes and death to 1</span>
      <span class="va">.</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"f"</span>, <span class="st">"no"</span>,  <span class="st">"Recover"</span><span class="op">)</span> <span class="op">~</span> <span class="fl">0</span>,           <span class="co">## female, no and recover to 0</span>
      <span class="cn">TRUE</span>                            <span class="op">~</span> <span class="cn">NA_real_</span><span class="op">)</span>    <span class="co">## otherwise set to missing</span>
    <span class="op">)</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="eliminar-las-filas-con-valores-faltantes" class="section level4 unnumbered">
<h4>Eliminar las filas con valores faltantes<a class="anchor" aria-label="anchor" href="#eliminar-las-filas-con-valores-faltantes"><i class="fas fa-link"></i></a>
</h4>
<p>Para eliminar las filas con valores faltantes, puede utilizar la función <code>drop_na()</code> de <strong>tidyr</strong>. Sin embargo, sólo queremos hacer esto para las filas a las que les faltan valores en las columnas de interés.</p>
<p>Lo primero que debemos hacer es asegurarnos de que nuestro vector <code>explanatory_vars</code> incluye la columna <code>age</code> (<code>age</code> habría producido un error en la operación anterior <code>case_when()</code>, que sólo era para variables dicotómicas). A continuación, canalizamos <code>linelist</code> a <code>drop_na()</code> para eliminar cualquier fila con valores faltantes en la columna <code>outcome</code> o en cualquiera de las columnas <code>explanatory_vars</code>.</p>
<p>Antes de ejecutar el código, el número de filas de <code>linelist</code> es <code><a href="https://rdrr.io/r/base/nrow.html">nrow(linelist)</a></code>.</p>
<div class="sourceCode" id="cb969"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## add in age_category to the explanatory vars </span>
<span class="va">explanatory_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">explanatory_vars</span>, <span class="st">"age_cat"</span><span class="op">)</span>

<span class="co">## drop rows with missing information for variables of interest </span>
<span class="va">linelist</span> <span class="op">&lt;-</span> <span class="va">linelist</span> <span class="op">%&gt;%</span> 
  <span class="fu">drop_na</span><span class="op">(</span><span class="fu">any_of</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"outcome"</span>, <span class="va">explanatory_vars</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>El número de filas que quedan en <code>linelist</code> es <code><a href="https://rdrr.io/r/base/nrow.html">nrow(linelist)</a></code>.</p>
<!-- ======================================================= -->
</div>
</div>
</div>
<div id="univariate" class="section level2" number="18.2">
<h2>
<span class="header-section-number">18.2</span> Univariante<a class="anchor" aria-label="anchor" href="#univariate"><i class="fas fa-link"></i></a>
</h2>
<p>Al igual que en la página sobre <a href="descriptive-tables.html#descriptive-tables">Tablas descriptivas</a>, tu caso de uso determinará el paquete de R que utilice. Presentamos dos opciones para realizar análisis univariantes:</p>
<ul>
<li><p>Utiliza las funciones disponibles en R <strong>base</strong> para imprimir rápidamente los resultados en la consola. Utiliza el paquete <strong>broom</strong> para ordenar las salidas.</p></li>
<li><p>Utiliza el paquete <strong>gtsummary</strong> para modelar y obtener resultados listos para tu publicación</p></li>
</ul>
<!-- ======================================================= --><div id="r-base-1" class="section level3 unnumbered">
<h3>R <strong>base</strong><a class="anchor" aria-label="anchor" href="#r-base-1"><i class="fas fa-link"></i></a>
</h3>
<div id="regresión-lineal" class="section level4 unnumbered">
<h4>Regresión lineal<a class="anchor" aria-label="anchor" href="#regresi%C3%B3n-lineal"><i class="fas fa-link"></i></a>
</h4>
<p>La función <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> de R <strong>base</strong>realiza una regresión lineal, evaluando la relación entre la respuesta numérica y las variables explicativas que se supone tienen una relación lineal.</p>
<p>Proporciona la ecuación como una fórmula, con los nombres de las columnas de respuesta y explicativa separados por una tilde <code><a href="https://rdrr.io/r/base/tilde.html">~</a></code>. Además, especifique los datos a <code>data =</code>. Defina los resultados del modelo como un objeto R, para utilizarlo más tarde.</p>
<div class="sourceCode" id="cb970"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">ht_cm</span> <span class="op">~</span> <span class="va">age</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span></code></pre></div>
<p>A continuación, puedes ejecutar <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> en los resultados del modelo para ver los coeficientes (estimaciones), el valor P, los residuos y otras medidas.</p>
<div class="sourceCode" id="cb971"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">lm_results</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = ht_cm ~ age, data = linelist)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -128.579  -15.854    1.177   15.887  175.483 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  69.9051     0.5979   116.9   &lt;2e-16 ***
## age           3.4354     0.0293   117.2   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 23.75 on 4165 degrees of freedom
## Multiple R-squared:  0.7675, Adjusted R-squared:  0.7674 
## F-statistic: 1.375e+04 on 1 and 4165 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>También se puede utilizar la función <code>tidy()</code> del paquete <strong>broom</strong> para obtener los resultados en una tabla. Lo que nos dicen los resultados es que por cada año de aumento de la edad la altura aumenta 3,5 cm y esto es estadísticamente significativo.</p>
<div class="sourceCode" id="cb973"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">tidy</span><span class="op">(</span><span class="va">lm_results</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 2 x 5
##   term        estimate std.error statistic p.value
##   &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;
## 1 (Intercept)    69.9     0.598       117.       0
## 2 age             3.44    0.0293      117.       0</code></pre>
<p>También puedes utilizar esta regresión para añadirla a un <strong>ggplot</strong>, para hacer esto, primero juntamos los puntos de los datos observados y la línea ajustada en un dataframe utilizando la función <code>augment()</code> de <strong>broom</strong>.</p>
<div class="sourceCode" id="cb975"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## pull the regression points and observed data in to one dataset</span>
<span class="va">points</span> <span class="op">&lt;-</span> <span class="fu">augment</span><span class="op">(</span><span class="va">lm_results</span><span class="op">)</span>

<span class="co">## plot the data using age as the x-axis </span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">points</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="co">## add points for height </span>
  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">ht_cm</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="co">## add your regression line </span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">.fitted</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/lin_reg_res_plot-1.png" width="672"></div>
<p>También es posible añadir una simple recta de regresión lineal en <strong>ggplot</strong> utilizando la función <code>geom_smooth()</code>.</p>
<div class="sourceCode" id="cb976"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## add your data to a plot </span>
 <span class="fu">ggplot</span><span class="op">(</span><span class="va">linelist</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">age</span>, y <span class="op">=</span> <span class="va">ht_cm</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="co">## show points</span>
  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="co">## add a linear regression </span>
  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"lm"</span>, se <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<div class="inline-figure"><img src="_main_files/figure-html/geom_smooth-1.png" width="672"></div>
<p>Consulta la sección de recursos al final de este capítulo para obtener tutoriales más detallados.</p>
</div>
<div id="regresión-logística" class="section level4 unnumbered">
<h4>Regresión logística<a class="anchor" aria-label="anchor" href="#regresi%C3%B3n-log%C3%ADstica"><i class="fas fa-link"></i></a>
</h4>
<p>La función <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> del paquete <strong>stats</strong> (parte de R <strong>base</strong>) se utiliza para ajustar los modelos lineales generalizados (GLM).</p>
<p><code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> puede utilizarse para la regresión logística univariante y multivariable (por ejemplo, para obtener Odds Ratios). Aquí están las partes principales:</p>
<div class="sourceCode" id="cb978"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># arguments for glm()</span>
<span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">formula</span>, <span class="va">family</span>, <span class="va">data</span>, <span class="va">weights</span>, <span class="va">subset</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
<ul>
<li>
<code>formula =</code> El modelo se proporciona a <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> como una ecuación, con el resultado a la izquierda y las variables explicativas a la derecha de una tilde ~.</li>
<li>
<code>family =</code> Determina el tipo de modelo a ejecutar. Para la regresión logística, utiliza <code>family = "binomial"</code>, para poisson utiliza <code>family = "poisson"</code>. Otros ejemplos se encuentran en la tabla siguiente.</li>
<li>
<code>data =</code> Especifica tu dataframe</li>
</ul>
<p>Si es necesario, también puede especificar la función de enlace mediante la sintaxis <code>family = familytype(link = "linkfunction"))</code>. Puedes leer más en la documentación sobre otras familias y argumentos opcionales como <code>weights =</code> y <code>subset =</code> (<code><a href="https://rdrr.io/r/stats/glm.html">?glm</a></code>).</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>Familia</th>
<th>Función de enlace por defecto</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>"binomial"</code></td>
<td><code>(link = "logit")</code></td>
</tr>
<tr class="even">
<td><code>"gaussian"</code></td>
<td><code>(link = "identity")</code></td>
</tr>
<tr class="odd">
<td><code>"Gamma"</code></td>
<td><code>(link = "inverse")</code></td>
</tr>
<tr class="even">
<td><code>"inverse.gaussian"</code></td>
<td><code>(link = "1/mu^2")</code></td>
</tr>
<tr class="odd">
<td><code>"poisson"</code></td>
<td><code>(link = "log")</code></td>
</tr>
<tr class="even">
<td><code>"quasi"</code></td>
<td><code>(link = "identity", variance = "constant")</code></td>
</tr>
<tr class="odd">
<td><code>"quasibinomial"</code></td>
<td><code>(link = "logit")</code></td>
</tr>
<tr class="even">
<td><code>"quasipoisson"</code></td>
<td><code>(link = "log")</code></td>
</tr>
</tbody>
</table></div>
<p>Cuando se ejecuta <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> lo más habitual es guardar los resultados como un objeto R. A continuación, puede imprimir los resultados en tu consola utilizando <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> como se muestra a continuación, o realizar otras operaciones con los resultados (por ejemplo, exponer).</p>
<p>Si necesitas ejecutar una regresión binomial negativa, puede utilizar el paquete <strong>MASS</strong>; glm.nb() utiliza la misma sintaxis que <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>.</p>
<p>Para un recorrido por diferentes regresiones, consulta la <a href="https://stats.idre.ucla.edu/other/dae/">página de estadísticas de UCLA</a>.</p>
</div>
<div id="univariante-glm" class="section level4 unnumbered">
<h4>Univariante <code>glm()</code><a class="anchor" aria-label="anchor" href="#univariante-glm"><i class="fas fa-link"></i></a>
</h4>
<p>En este ejemplo estamos evaluando la asociación entre diferentes categorías de edad y el resultado de muerte (codificado como 1 en la sección Preparación). A continuación se muestra un modelo univariante de <code>outcome</code> por age_cat. Guardamos la salida del modelo como <code>model</code> y luego la imprimimos con <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> en la consola. Observe que las estimaciones proporcionadas son las <em>probabilidades logarítmicas (log odds)</em> y que el nivel de referencia es el primer nivel del factor <code>age_cat</code> (“0-4”).</p>
<div class="sourceCode" id="cb979"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">age_cat</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">model</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = outcome ~ age_cat, family = "binomial", data = linelist)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -1.339  -1.278   1.024   1.080   1.354  
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)   
## (Intercept)   0.233738   0.072805   3.210  0.00133 **
## age_cat5-9   -0.062898   0.101733  -0.618  0.53640   
## age_cat10-14  0.138204   0.107186   1.289  0.19726   
## age_cat15-19 -0.005565   0.113343  -0.049  0.96084   
## age_cat20-29  0.027511   0.102133   0.269  0.78765   
## age_cat30-49  0.063764   0.113771   0.560  0.57517   
## age_cat50-69 -0.387889   0.259240  -1.496  0.13459   
## age_cat70+   -0.639203   0.915770  -0.698  0.48518   
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 5712.4  on 4166  degrees of freedom
## Residual deviance: 5705.1  on 4159  degrees of freedom
## AIC: 5721.1
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>Para modificar el nivel de referencia de una variable determinada, asegúrese de que la columna es del tipo Factor y mueva el nivel deseado a la primera posición con <code>fct_relevel()</code> (véase la página sobre <a href="#factors">Factores</a>). Por ejemplo, a continuación tomamos la columna <code>age_cat</code> y establecemos “20-29” como línea de base antes de canalizar el dataframe modificado en <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>.</p>
<div class="sourceCode" id="cb981"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">linelist</span> <span class="op">%&gt;%</span> 
  <span class="fu">mutate</span><span class="op">(</span>age_cat <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span><span class="va">age_cat</span>, <span class="st">"20-29"</span>, after <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="va">outcome</span> <span class="op">~</span> <span class="va">age_cat</span>, family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = outcome ~ age_cat, family = "binomial", data = .)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -1.339  -1.278   1.024   1.080   1.354  
## 
## Coefficients:
##              Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   0.26125    0.07163   3.647 0.000265 ***
## age_cat0-4   -0.02751    0.10213  -0.269 0.787652    
## age_cat5-9   -0.09041    0.10090  -0.896 0.370220    
## age_cat10-14  0.11069    0.10639   1.040 0.298133    
## age_cat15-19 -0.03308    0.11259  -0.294 0.768934    
## age_cat30-49  0.03625    0.11302   0.321 0.748390    
## age_cat50-69 -0.41540    0.25891  -1.604 0.108625    
## age_cat70+   -0.66671    0.91568  -0.728 0.466546    
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 5712.4  on 4166  degrees of freedom
## Residual deviance: 5705.1  on 4159  degrees of freedom
## AIC: 5721.1
## 
## Number of Fisher Scoring iterations: 4</code></pre>
</div>
<div id="resultados-de-la-impresión" class="section level4 unnumbered">
<h4>Resultados de la impresión<a class="anchor" aria-label="anchor" href="#resultados-de-la-impresi%C3%B3n"><i class="fas fa-link"></i></a>
</h4>
<p>Para la mayoría de los usos, hay que hacer varias modificaciones a los resultados anteriores. La función <code>tidy()</code> del paquete <strong>broom</strong> es conveniente para hacer presentables los resultados del modelo.</p>
<p>Aquí demostramos cómo combinar los resultados del modelo con una tabla de recuentos.</p>
<ol style="list-style-type: decimal">
<li>Obtenga las estimaciones de log odds ratio <em>exponenciadas</em> y los intervalos de confianza pasando el modelo a <code>tidy()</code> y estableciendo <code>exponentiate = TRUE</code> y <code>conf.int = TRUE</code>.</li>
</ol>
<div class="sourceCode" id="cb983"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">age_cat</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tidy</span><span class="op">(</span>exponentiate <span class="op">=</span> <span class="cn">TRUE</span>, conf.int <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>        <span class="co"># exponentiate and produce CIs</span>
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span>, <span class="va">round</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>  <span class="co"># round all numeric columns</span></code></pre></div>
<p>A continuación, se muestra el <code>model</code> tibble resultante:</p>
<div id="htmlwidget-1e525960738762c31fc9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1e525960738762c31fc9">{"x":{"filter":"none","vertical":false,"data":[["(Intercept)","age_cat5-9","age_cat10-14","age_cat15-19","age_cat20-29","age_cat30-49","age_cat50-69","age_cat70+"],[1.26,0.94,1.15,0.99,1.03,1.07,0.68,0.53],[0.07,0.1,0.11,0.11,0.1,0.11,0.26,0.92],[3.21,-0.62,1.29,-0.05,0.27,0.56,-1.5,-0.7],[0,0.54,0.2,0.96,0.79,0.58,0.13,0.49],[1.1,0.77,0.93,0.8,0.84,0.85,0.41,0.07],[1.46,1.15,1.42,1.24,1.26,1.33,1.13,3.2]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>std.error<\/th>\n      <th>statistic<\/th>\n      <th>p.value<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":8,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[8,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><ol start="2" style="list-style-type: decimal">
<li>Combina estos resultados del modelo con una tabla de recuentos. A continuación, creamos la tabla cruzada de recuentos con la función <code>tabyl()</code> de <strong>janitor</strong>, como se explica en la página de <a href="descriptive-tables.html#descriptive-tables">tablas descriptivas</a>.</li>
</ol>
<div class="sourceCode" id="cb984"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_table</span> <span class="op">&lt;-</span> <span class="va">linelist</span> <span class="op">%&gt;%</span> 
  <span class="fu">janitor</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/janitor/man/tabyl.html">tabyl</a></span><span class="op">(</span><span class="va">age_cat</span>, <span class="va">outcome</span><span class="op">)</span></code></pre></div>
<!-- * Group rows by outcome, and get counts by age category   -->
<!-- * Pivot wider so the column are `age_cat`, `0`, and `1`   -->
<!-- * Remove row for `NA` `age_cat`, if applicable, to align with the model results   -->
<!-- ```{r} -->
<!-- counts_table <- linelist %>%  -->
<!--   filter(!is.na(outcome) & !is.na(age_cat)) %>%    # ensure outcome and age_cat are present  -->
<!--   group_by(outcome) %>%                            # get counts of variable of interest grouped by outcome -->
<!--   count(age_cat) %>%   ## gets number or rows by unique outcome-age category combinations   -->
<!--   pivot_wider(names_from = outcome, values_from = n)    ## spread data to wide format (as in cross-tabulation) -->
<!-- ``` -->
<p>Este es el aspecto de este dataframe <code>counts_table</code>:</p>
<div id="htmlwidget-ed87d33b76b078208bdf" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ed87d33b76b078208bdf">{"x":{"filter":"none","vertical":false,"data":[["0-4","5-9","10-14","15-19","20-29","30-49","50-69","70+"],[338,365,273,238,345,228,35,3],[427,433,396,299,448,307,30,2]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>age_cat<\/th>\n      <th>0<\/th>\n      <th>1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":8,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[8,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><p>Ahora podemos unir <code>counts_table</code> y los resultados del <code>model</code> horizontalmente con <code>bind_cols()</code> (<strong>dplyr</strong>). Recuerda que con <code>bind_cols()</code> las filas de los dos dataframes deben estar perfectamente alineadas. En este código, como estamos enlazando dentro de una cadena de tuberías, utilizamos <code>.</code> para representar el objeto de tuberías <code>counts_table</code> mientras lo enlazamos con el modelo. Para terminar el proceso, utilizamos <code>select()</code> para elegir las columnas deseadas y tu orden, y finalmente aplicamos la función <code><a href="https://rdrr.io/r/base/Round.html">round()</a></code> de R <strong>base</strong> en todas las columnas numéricas para especificar 2 decimales.</p>
<div class="sourceCode" id="cb985"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">combined</span> <span class="op">&lt;-</span> <span class="va">counts_table</span> <span class="op">%&gt;%</span>           <span class="co"># begin with table of counts</span>
  <span class="fu">bind_cols</span><span class="op">(</span><span class="va">.</span>, <span class="va">model</span><span class="op">)</span> <span class="op">%&gt;%</span>              <span class="co"># combine with the outputs of the regression </span>
  <span class="fu">select</span><span class="op">(</span><span class="va">term</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="va">estimate</span>,          <span class="co"># select and re-order cols</span>
         <span class="va">conf.low</span>, <span class="va">conf.high</span>, <span class="va">p.value</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span>, <span class="va">round</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co">## round to 2 decimal places</span></code></pre></div>
<p>Este es el aspecto del dataframe combinado, impreso de forma agradable como una imagen con una función de <strong>flextable</strong>. En <a href="#tables-for-presentation">Tablas para presentación</a> se explica cómo personalizar dichas tablas con <strong>flextable</strong>, o bien puede utilizar otros numerosos paquetes como <strong>knitr</strong> o <strong>GT</strong>.</p>
<div class="sourceCode" id="cb986"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">combined</span> <span class="op">&lt;-</span> <span class="va">combined</span> <span class="op">%&gt;%</span> 
  <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">qflextable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="bucle-de-múltiples-modelos-univariantes" class="section level4 unnumbered">
<h4>Bucle de múltiples modelos univariantes<a class="anchor" aria-label="anchor" href="#bucle-de-m%C3%BAltiples-modelos-univariantes"><i class="fas fa-link"></i></a>
</h4>
<p>A continuación presentamos un método que utiliza <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> y <code>tidy()</code> para un enfoque más sencillo, véase la sección sobre <strong>gtsummary</strong>.</p>
<p>Para ejecutar los modelos en varias variables de exposición para producir odds ratios univariantes (es decir, sin controlar entre sí), puede utilizar el enfoque siguiente. Utiliza <code>str_c()</code> de <strong>stringr</strong> para crear fórmulas univariantes (véase <a href="characters-and-strings.html#characters-and-strings">Caracteres y cadenas</a>), ejecuta la regresión <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> en cada fórmula, pasa cada resultado de <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> a <code>tidy()</code> y finalmente colapsa todos los resultados del modelo juntos con <code>bind_rows()</code> de <strong>tidyr</strong>. Este enfoque utiliza <code>map()</code> del paquete <strong>purrr</strong> para iterar - véase la página sobre <a href="iteration-loops-and-lists.html#iteration-loops-and-lists">Iteración, bucles y listas</a> para más información sobre esta herramienta.</p>
<ol style="list-style-type: decimal">
<li><p>crea un vector de nombres de columnas de las variables explicativas. Ya lo tenemos como <code>explanatory_vars</code> de la sección de preparación de esta página.</p></li>
<li><p>Utiliza <code>str_c()</code> para crear múltiples fórmulas de cadena, con el resultado a la izquierda, y un nombre de columna de <code>explanatory_vars</code> a la derecha. El punto <code>.</code> sustituye al nombre de la columna en <code>explanatory_vars</code>.</p></li>
</ol>
<div class="sourceCode" id="cb987"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">explanatory_vars</span> <span class="op">%&gt;%</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">"outcome ~ "</span>, <span class="va">.</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "outcome ~ gender"  "outcome ~ fever"   "outcome ~ chills" 
## [4] "outcome ~ cough"   "outcome ~ aches"   "outcome ~ vomit"  
## [7] "outcome ~ age_cat"</code></pre>
<ol start="3" style="list-style-type: decimal">
<li><p>Pasa estas fórmulas de cadena a <code>map()</code> y establece <code>~glm()</code> como la función a aplicar a cada entrada. Dentro de <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>, establece la fórmula de regresión como <code><a href="https://rdrr.io/r/stats/formula.html">as.formula(.x)</a></code>, donde <code>.x</code> se sustituirá por la fórmula de cadena definida en el paso anterior. <code>map()</code> realizará un bucle sobre cada una de las fórmulas de cadena, ejecutando regresiones para cada una.</p></li>
<li><p>Los resultados de este primer <code>map()</code> se pasan a un segundo comando <code>map()</code>, que aplica <code>tidy()</code> a los resultados de la regresión.</p></li>
<li><p>Por último, la salida de la segunda función <code>map()</code> (una lista de dataframes ordenados) se condensa con <code>bind_rows()</code>, dando lugar a un dataframe con todos los resultados univariantes.</p></li>
</ol>
<div class="sourceCode" id="cb989"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">models</span> <span class="op">&lt;-</span> <span class="va">explanatory_vars</span> <span class="op">%&gt;%</span>       <span class="co"># begin with variables of interest</span>
  <span class="fu">str_c</span><span class="op">(</span><span class="st">"outcome ~ "</span>, <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>         <span class="co"># combine each variable into formula ("outcome ~ variable of interest")</span>
  
  <span class="co"># iterate through each univariate formula</span>
  <span class="fu">map</span><span class="op">(</span>                               
    .f <span class="op">=</span> <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>                       <span class="co"># pass the formulas one-by-one to glm()</span>
      formula <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>,      <span class="co"># within glm(), the string formula is .x</span>
      family <span class="op">=</span> <span class="st">"binomial"</span>,           <span class="co"># specify type of glm (logistic)</span>
      data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>          <span class="co"># dataset</span>
  
  <span class="co"># tidy up each of the glm regression outputs from above</span>
  <span class="fu">map</span><span class="op">(</span>
    .f <span class="op">=</span> <span class="op">~</span><span class="fu">tidy</span><span class="op">(</span>
      <span class="va">.x</span>, 
      exponentiate <span class="op">=</span> <span class="cn">TRUE</span>,           <span class="co"># exponentiate </span>
      conf.int <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>          <span class="co"># return confidence intervals</span>
  
  <span class="co"># collapse the list of regression outputs in to one data frame</span>
  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co"># round all numeric columns</span>
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span>, <span class="va">round</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Esta vez, el objeto final <code>models</code> es más largo porque ahora representa los resultados combinados de varias regresiones univariantes. Clica para ver todas las filas de <code>model</code>.</p>
<div id="htmlwidget-221ca619ed4af3ec9f7c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-221ca619ed4af3ec9f7c">{"x":{"filter":"none","vertical":false,"data":[["(Intercept)","gender","(Intercept)","fever","(Intercept)","chills","(Intercept)","cough","(Intercept)","aches","(Intercept)","vomit","(Intercept)","age_cat5-9","age_cat10-14","age_cat15-19","age_cat20-29","age_cat30-49","age_cat50-69","age_cat70+"],[1.28,1,1.28,1,1.28,1.03,1.14,1.15,1.29,0.93,1.23,1.09,1.26,0.94,1.15,0.99,1.03,1.07,0.68,0.53],[0.04,0.06,0.07,0.08,0.03,0.08,0.08,0.09,0.03,0.1,0.04,0.06,0.07,0.1,0.11,0.11,0.1,0.11,0.26,0.92],[5.67,-0.04,3.44,0.01,6.98,0.41,1.53,1.58,7.78,-0.66,4.67,1.39,3.21,-0.62,1.29,-0.05,0.27,0.56,-1.5,-0.7],[0,0.97,0,0.99,0,0.68,0.13,0.11,0,0.51,0,0.17,0,0.54,0.2,0.96,0.79,0.58,0.13,0.49],[1.18,0.88,1.11,0.85,1.19,0.89,0.97,0.97,1.21,0.76,1.13,0.96,1.1,0.77,0.93,0.8,0.84,0.85,0.41,0.07],[1.4,1.13,1.48,1.17,1.37,1.21,1.34,1.37,1.38,1.14,1.34,1.23,1.46,1.15,1.42,1.24,1.26,1.33,1.13,3.2]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>std.error<\/th>\n      <th>statistic<\/th>\n      <th>p.value<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><p>Como antes, podemos crear una tabla de recuentos a partir de <code>linelist</code> para cada variable explicativa, vincularla a <code>models</code> y hacer una bonita tabla. Comenzamos con las variables, e iteramos a través de ellas con <code>map()</code>. Iteramos a través de una función definida por el usuario que implica la creación de una tabla de recuentos con funciones <strong>dplyr</strong>. Luego se combinan los resultados y se vinculan con los resultados del modelo <code>models</code>.</p>
<div class="sourceCode" id="cb990"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## for each explanatory variable</span>
<span class="va">univ_tab_base</span> <span class="op">&lt;-</span> <span class="va">explanatory_vars</span> <span class="op">%&gt;%</span> 
  <span class="fu">map</span><span class="op">(</span>.f <span class="op">=</span> 
    <span class="op">~</span><span class="op">{</span><span class="va">linelist</span> <span class="op">%&gt;%</span>                <span class="co">## begin with linelist</span>
        <span class="fu">group_by</span><span class="op">(</span><span class="va">outcome</span><span class="op">)</span> <span class="op">%&gt;%</span>     <span class="co">## group data set by outcome</span>
        <span class="fu">count</span><span class="op">(</span><span class="va">.data</span><span class="op">[[</span><span class="va">.x</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span>    <span class="co">## produce counts for variable of interest</span>
        <span class="fu">pivot_wider</span><span class="op">(</span>              <span class="co">## spread to wide format (as in cross-tabulation)</span>
          names_from <span class="op">=</span> <span class="va">outcome</span>,
          values_from <span class="op">=</span> <span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> 
        <span class="fu">drop_na</span><span class="op">(</span><span class="va">.data</span><span class="op">[[</span><span class="va">.x</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span>         <span class="co">## drop rows with missings</span>
        <span class="fu">rename</span><span class="op">(</span><span class="st">"variable"</span> <span class="op">=</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span>      <span class="co">## change variable of interest column to "variable"</span>
        <span class="fu">mutate</span><span class="op">(</span>variable <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span><span class="op">)</span><span class="op">}</span> <span class="co">## convert to character, else non-dichotomous (categorical) variables come out as factor and cant be merged</span>
      <span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co">## collapse the list of count outputs in to one data frame</span>
  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co">## merge with the outputs of the regression </span>
  <span class="fu">bind_cols</span><span class="op">(</span><span class="va">.</span>, <span class="va">models</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co">## only keep columns interested in </span>
  <span class="fu">select</span><span class="op">(</span><span class="va">term</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="va">estimate</span>, <span class="va">conf.low</span>, <span class="va">conf.high</span>, <span class="va">p.value</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co">## round decimal places</span>
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span>, <span class="va">round</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>A continuación se muestra el aspecto del dataframe. Consulta la página sobre <a href="#tables-for-presentation">Tablas para presentación</a> para obtener ideas sobre cómo convertir esta tabla en una bonita salida HTML (por ejemplo, con <strong>flextable</strong>).</p>
<div id="htmlwidget-c1090d5642f8aed1245f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c1090d5642f8aed1245f">{"x":{"filter":"none","vertical":false,"data":[["(Intercept)","gender","(Intercept)","fever","(Intercept)","chills","(Intercept)","cough","(Intercept)","aches","(Intercept)","vomit","(Intercept)","age_cat5-9","age_cat10-14","age_cat15-19","age_cat20-29","age_cat30-49","age_cat50-69","age_cat70+"],[909,916,340,1485,1472,353,272,1553,1636,189,931,894,338,365,273,238,345,228,35,3],[1168,1174,436,1906,1877,465,309,2033,2114,228,1144,1198,427,433,396,299,448,307,30,2],[1.28,1,1.28,1,1.28,1.03,1.14,1.15,1.29,0.93,1.23,1.09,1.26,0.94,1.15,0.99,1.03,1.07,0.68,0.53],[1.18,0.88,1.11,0.85,1.19,0.89,0.97,0.97,1.21,0.76,1.13,0.96,1.1,0.77,0.93,0.8,0.84,0.85,0.41,0.07],[1.4,1.13,1.48,1.17,1.37,1.21,1.34,1.37,1.38,1.14,1.34,1.23,1.46,1.15,1.42,1.24,1.26,1.33,1.13,3.2],[0,0.97,0,0.99,0,0.68,0.13,0.11,0,0.51,0,0.17,0,0.54,0.2,0.96,0.79,0.58,0.13,0.49]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>term<\/th>\n      <th>0<\/th>\n      <th>1<\/th>\n      <th>estimate<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n      <th>p.value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><!-- ======================================================= -->
</div>
</div>
<div id="reg_gt_uni" class="section level3 unnumbered">
<h3>paquete <strong>gtsummary</strong><a class="anchor" aria-label="anchor" href="#reg_gt_uni"><i class="fas fa-link"></i></a>
</h3>
<p>A continuación presentamos el uso de <code>tbl_uvregression()</code> del paquete <strong>gtsummary</strong>. Al igual que en la página sobre <a href="descriptive-tables.html#descriptive-tables">Tablas descriptivas</a>, las funciones de gtsummary hacen un buen trabajo al ejecutar estadísticas <em>y</em> producir salidas de aspecto profesional. Esta función produce una tabla de resultados de regresión univariante.</p>
<p>Seleccionamos sólo las columnas necesarias de <code>linelist</code> (variables explicativas y la variable de resultado) y las introducimos en <code>tbl_uvregression()</code>. Vamos a ejecutar una regresión univariante en cada una de las columnas que definimos como <code>explanatory_vars</code> en la sección de preparación de datos (sexo, fiebre, escalofríos, tos, dolores, vómitos y age_cat).</p>
<p>Dentro de la propia función, proporcionamos el <code>method =</code> como <code>glm</code> (sin comillas), la columna de resultado <code>y =</code> (<code>outcome</code>), especificamos a <code>method.args =</code> que queremos ejecutar la regresión logística a través de <code>family = binomial</code>, y le decimos que exponencie los resultados.</p>
<p>La salida es HTML y contiene los recuentos</p>
<div class="sourceCode" id="cb991"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">univ_tab</span> <span class="op">&lt;-</span> <span class="va">linelist</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">explanatory_vars</span>, <span class="va">outcome</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co">## select variables of interest</span>

  <span class="fu">tbl_uvregression</span><span class="op">(</span>                         <span class="co">## produce univariate table</span>
    method <span class="op">=</span> <span class="va">glm</span>,                           <span class="co">## define regression want to run (generalised linear model)</span>
    y <span class="op">=</span> <span class="va">outcome</span>,                            <span class="co">## define outcome variable</span>
    method.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>family <span class="op">=</span> <span class="va">binomial</span><span class="op">)</span>,  <span class="co">## define what type of glm want to run (logistic)</span>
    exponentiate <span class="op">=</span> <span class="cn">TRUE</span>                     <span class="co">## exponentiate to produce odds ratios (rather than log odds)</span>
  <span class="op">)</span>

<span class="co">## view univariate results table </span>
<span class="va">univ_tab</span></code></pre></div>
<div id="hciokcklgd" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#hciokcklgd .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#hciokcklgd .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#hciokcklgd .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#hciokcklgd .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#hciokcklgd .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hciokcklgd .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#hciokcklgd .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#hciokcklgd .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#hciokcklgd .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#hciokcklgd .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#hciokcklgd .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#hciokcklgd .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#hciokcklgd .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#hciokcklgd .gt_from_md > :first-child {
  margin-top: 0;
}

#hciokcklgd .gt_from_md > :last-child {
  margin-bottom: 0;
}

#hciokcklgd .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#hciokcklgd .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#hciokcklgd .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#hciokcklgd .gt_row_group_first td {
  border-top-width: 2px;
}

#hciokcklgd .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#hciokcklgd .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#hciokcklgd .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#hciokcklgd .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hciokcklgd .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#hciokcklgd .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#hciokcklgd .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#hciokcklgd .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hciokcklgd .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#hciokcklgd .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#hciokcklgd .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#hciokcklgd .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#hciokcklgd .gt_left {
  text-align: left;
}

#hciokcklgd .gt_center {
  text-align: center;
}

#hciokcklgd .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#hciokcklgd .gt_font_normal {
  font-weight: normal;
}

#hciokcklgd .gt_font_bold {
  font-weight: bold;
}

#hciokcklgd .gt_font_italic {
  font-style: italic;
}

#hciokcklgd .gt_super {
  font-size: 65%;
}

#hciokcklgd .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#hciokcklgd .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#hciokcklgd .gt_indent_1 {
  text-indent: 5px;
}

#hciokcklgd .gt_indent_2 {
  text-indent: 10px;
}

#hciokcklgd .gt_indent_3 {
  text-indent: 15px;
}

#hciokcklgd .gt_indent_4 {
  text-indent: 20px;
}

#hciokcklgd .gt_indent_5 {
  text-indent: 25px;
}
</style>
<div class="inline-table"><table class="gt_table">
<thead class="gt_col_headings"><tr>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>N</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>OR</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>95% CI</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>p-value</strong></th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td class="gt_row gt_left">gender</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.88, 1.13</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">fever</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.85, 1.17</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">chills</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.89, 1.21</td>
<td class="gt_row gt_center">0.7</td>
</tr>
<tr>
<td class="gt_row gt_left">cough</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.97, 1.37</td>
<td class="gt_row gt_center">0.11</td>
</tr>
<tr>
<td class="gt_row gt_left">aches</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">0.93</td>
<td class="gt_row gt_center">0.76, 1.14</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left">vomit</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.09</td>
<td class="gt_row gt_center">0.96, 1.23</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left">age_cat</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">0-4</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">5-9</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.94</td>
<td class="gt_row gt_center">0.77, 1.15</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">10-14</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.93, 1.42</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">15-19</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.99</td>
<td class="gt_row gt_center">0.80, 1.24</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">20-29</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.84, 1.26</td>
<td class="gt_row gt_center">0.8</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">30-49</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.07</td>
<td class="gt_row gt_center">0.85, 1.33</td>
<td class="gt_row gt_center">0.6</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">50-69</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.68</td>
<td class="gt_row gt_center">0.41, 1.13</td>
<td class="gt_row gt_center">0.13</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">70+</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.53</td>
<td class="gt_row gt_center">0.07, 3.20</td>
<td class="gt_row gt_center">0.5</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="5">
<sup class="gt_footnote_marks">1</sup> OR = Odds Ratio, CI = Confidence Interval</td>
    </tr></tfoot>
</table></div>
</div>
<p>Hay muchas modificaciones que se pueden hacer a la salida de esta tabla, como ajustar las etiquetas de texto, poner en negrita las filas por tu valor p, etc. Consulta <a href="http://www.danieldsjoberg.com/gtsummary/articles/tbl_regression.html">estos tutoriales</a> y en otros lugares en línea.</p>
<!-- ======================================================= -->
</div>
</div>
<div id="stratified" class="section level2" number="18.3">
<h2>
<span class="header-section-number">18.3</span> Estratificado<a class="anchor" aria-label="anchor" href="#stratified"><i class="fas fa-link"></i></a>
</h2>
<p>El análisis estratificado se está trabajando actualmente para <strong>gtsummary</strong>, esta página se actualizará a su debido tiempo.</p>
</div>
<div id="multivariable" class="section level2" number="18.4">
<h2>
<span class="header-section-number">18.4</span> Multivariable<a class="anchor" aria-label="anchor" href="#multivariable"><i class="fas fa-link"></i></a>
</h2>
<p>Para el análisis multivariable, volvemos a presentar dos enfoques:</p>
<ul>
<li>
<code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> y <code>tidy()</code><br>
</li>
<li>paquete <strong>gtsummary</strong>
</li>
</ul>
<p>El flujo de trabajo es similar para cada uno de ellos y sólo el último paso de elaborar una tabla final es diferente.</p>
<div id="realizar-el-multivariable" class="section level3 unnumbered">
<h3>Realizar el multivariable<a class="anchor" aria-label="anchor" href="#realizar-el-multivariable"><i class="fas fa-link"></i></a>
</h3>
<p>Aquí utilizamos <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> pero añadimos más variables al lado derecho de la ecuación, separadas por símbolos de suma (<code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code>).</p>
<p>Para ejecutar el modelo con todas nuestras variables explicativas ejecutaríamos:</p>
<div class="sourceCode" id="cb992"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mv_reg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">gender</span> <span class="op">+</span> <span class="va">fever</span> <span class="op">+</span> <span class="va">chills</span> <span class="op">+</span> <span class="va">cough</span> <span class="op">+</span> <span class="va">aches</span> <span class="op">+</span> <span class="va">vomit</span> <span class="op">+</span> <span class="va">age_cat</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mv_reg</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = outcome ~ gender + fever + chills + cough + aches + 
##     vomit + age_cat, family = "binomial", data = linelist)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -1.383  -1.279   1.029   1.078   1.346  
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)
## (Intercept)   0.069054   0.131726   0.524    0.600
## gender        0.002448   0.065133   0.038    0.970
## fever         0.004309   0.080522   0.054    0.957
## chills        0.034112   0.078924   0.432    0.666
## cough         0.138584   0.089909   1.541    0.123
## aches        -0.070705   0.104078  -0.679    0.497
## vomit         0.086098   0.062618   1.375    0.169
## age_cat5-9   -0.063562   0.101851  -0.624    0.533
## age_cat10-14  0.136372   0.107275   1.271    0.204
## age_cat15-19 -0.011074   0.113640  -0.097    0.922
## age_cat20-29  0.026552   0.102780   0.258    0.796
## age_cat30-49  0.059569   0.116402   0.512    0.609
## age_cat50-69 -0.388964   0.262384  -1.482    0.138
## age_cat70+   -0.647443   0.917375  -0.706    0.480
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 5712.4  on 4166  degrees of freedom
## Residual deviance: 5700.2  on 4153  degrees of freedom
## AIC: 5728.2
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>Si quieres incluir dos variables y una interacción entre ellas puede separarlas con un asterisco <code><a href="https://rdrr.io/r/base/Arithmetic.html">*</a></code> en lugar de un <code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code>. Si sólo especifica la interacción, sepárelas con dos puntos <code><a href="https://rdrr.io/r/base/Colon.html">:</a></code>. Por ejemplo:</p>
<div class="sourceCode" id="cb994"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">gender</span> <span class="op">+</span> <span class="va">age_cat</span> <span class="op">*</span> <span class="va">fever</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span></code></pre></div>
<p><em>Opcionalmente</em>, puedes utilizar este código para aprovechar el vector predefinido de nombres de columnas y volver a crear el comando anterior utilizando <code>str_c()</code>. Esto puede ser útil si los nombres de sus variables explicativas cambian, o si no quiere escribirlos todos de nuevo.</p>
<div class="sourceCode" id="cb995"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## run a regression with all variables of interest </span>
<span class="va">mv_reg</span> <span class="op">&lt;-</span> <span class="va">explanatory_vars</span> <span class="op">%&gt;%</span>  <span class="co">## begin with vector of explanatory column names</span>
  <span class="fu">str_c</span><span class="op">(</span>collapse <span class="op">=</span> <span class="st">"+"</span><span class="op">)</span> <span class="op">%&gt;%</span>     <span class="co">## combine all names of the variables of interest separated by a plus</span>
  <span class="fu">str_c</span><span class="op">(</span><span class="st">"outcome ~ "</span>, <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>    <span class="co">## combine the names of variables of interest with outcome in formula style</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"binomial"</span>,      <span class="co">## define type of glm as logistic,</span>
      data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span>          <span class="co">## define your dataset</span></code></pre></div>
<div id="construir-el-modelo" class="section level4 unnumbered">
<h4>Construir el modelo<a class="anchor" aria-label="anchor" href="#construir-el-modelo"><i class="fas fa-link"></i></a>
</h4>
<p>Puedes construir tu modelo paso a paso, guardando varios modelos que incluyan determinadas variables explicativas. Puedes comparar estos modelos con pruebas de razón de verosimilitud utilizando `lrtest() del paquete <strong>lmtest</strong>, como se indica a continuación:</p>
<p><span style="color: black;"><strong><em>NOTA:</em></strong> El uso de anova(model1, model2, test = “Chisq) de R <strong>base</strong> produce los mismos resultados </span></p>
<div class="sourceCode" id="cb996"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">age_cat</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span>
<span class="va">model2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">outcome</span> <span class="op">~</span> <span class="va">age_cat</span> <span class="op">+</span> <span class="va">gender</span>, family <span class="op">=</span> <span class="st">"binomial"</span>, data <span class="op">=</span> <span class="va">linelist</span><span class="op">)</span>

<span class="fu">lmtest</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/lmtest/man/lrtest.html">lrtest</a></span><span class="op">(</span><span class="va">model1</span>, <span class="va">model2</span><span class="op">)</span></code></pre></div>
<pre><code>## Likelihood ratio test
## 
## Model 1: outcome ~ age_cat
## Model 2: outcome ~ age_cat + gender
##   #Df  LogLik Df  Chisq Pr(&gt;Chisq)
## 1   8 -2852.6                     
## 2   9 -2852.6  1 0.0002     0.9883</code></pre>
<p>Otra opción es tomar el objeto modelo y aplicar la función <code><a href="https://rdrr.io/r/stats/step.html">step()</a></code> del paquete <strong>stats</strong>. Especifica qué dirección de selección de variables deseas utilizar al construir el modelo.</p>
<div class="sourceCode" id="cb998"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## choose a model using forward selection based on AIC</span>
<span class="co">## you can also do "backward" or "both" by adjusting the direction</span>
<span class="va">final_mv_reg</span> <span class="op">&lt;-</span> <span class="va">mv_reg</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/step.html">step</a></span><span class="op">(</span>direction <span class="op">=</span> <span class="st">"forward"</span>, trace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>También puedes desactivar la notación científica en tu sesión de R, para mayor claridad:</p>
<div class="sourceCode" id="cb999"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>scipen<span class="op">=</span><span class="fl">999</span><span class="op">)</span></code></pre></div>
<p>Como se describe en la sección sobre el análisis univariante, pasamos la salida del modelo a <code>tidy()</code> para exponer las probabilidades logarítmicas y los IC. Finalmente, redondeamos todas las columnas numéricas a dos decimales. Desplázate para ver todas las filas.</p>
<div class="sourceCode" id="cb1000"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mv_tab_base</span> <span class="op">&lt;-</span> <span class="va">final_mv_reg</span> <span class="op">%&gt;%</span> 
  <span class="fu">broom</span><span class="fu">::</span><span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html">tidy</a></span><span class="op">(</span>exponentiate <span class="op">=</span> <span class="cn">TRUE</span>, conf.int <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="co">## get a tidy dataframe of estimates </span>
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span>, <span class="va">round</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>          <span class="co">## round </span></code></pre></div>
<p>Este es el aspecto del dataframe resultante:</p>
<div id="htmlwidget-7f16e72b21acf0a0cdd5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7f16e72b21acf0a0cdd5">{"x":{"filter":"none","vertical":false,"data":[["(Intercept)","gender","fever","chills","cough","aches","vomit","age_cat5-9","age_cat10-14","age_cat15-19","age_cat20-29","age_cat30-49","age_cat50-69","age_cat70+"],[1.07,1,1,1.03,1.15,0.93,1.09,0.94,1.15,0.99,1.03,1.06,0.68,0.52],[0.13,0.07,0.08,0.08,0.09,0.1,0.06,0.1,0.11,0.11,0.1,0.12,0.26,0.92],[0.52,0.04,0.05,0.43,1.54,-0.68,1.37,-0.62,1.27,-0.1,0.26,0.51,-1.48,-0.71],[0.6,0.97,0.96,0.67,0.12,0.5,0.17,0.53,0.2,0.92,0.8,0.61,0.14,0.48],[0.83,0.88,0.86,0.89,0.96,0.76,0.96,0.77,0.93,0.79,0.84,0.85,0.4,0.07],[1.39,1.14,1.18,1.21,1.37,1.14,1.23,1.15,1.41,1.24,1.26,1.33,1.13,3.19]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>std.error<\/th>\n      <th>statistic<\/th>\n      <th>p.value<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><!-- ======================================================= -->
</div>
</div>
<div id="combine-univariate-and-multivariable" class="section level3 unnumbered">
<h3>Combine univariate and multivariable<a class="anchor" aria-label="anchor" href="#combine-univariate-and-multivariable"><i class="fas fa-link"></i></a>
</h3>
<div id="combinar-con-gtsummary" class="section level4 unnumbered">
<h4>Combinar con <strong>gtsummary</strong><a class="anchor" aria-label="anchor" href="#combinar-con-gtsummary"><i class="fas fa-link"></i></a>
</h4>
<p>El paquete <strong>gtsummary</strong> proporciona la función <code>tbl_regression()</code>, que tomará los resultados de una regresión (<code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> en este caso) y producirá una bonita tabla resumen.</p>
<div class="sourceCode" id="cb1001"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## show results table of final regression </span>
<span class="va">mv_tab</span> <span class="op">&lt;-</span> <span class="fu">tbl_regression</span><span class="op">(</span><span class="va">final_mv_reg</span>, exponentiate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## ! Use of `vars()` is now deprecated and support will soon be removed. Please replace calls to `vars()` with `c()`.</code></pre>
<p>Veamos la tabla:</p>
<div class="sourceCode" id="cb1003"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mv_tab</span></code></pre></div>
<pre><code>## Warning: Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.</code></pre>
<div id="wmwobwlzqj" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#wmwobwlzqj .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#wmwobwlzqj .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#wmwobwlzqj .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#wmwobwlzqj .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#wmwobwlzqj .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#wmwobwlzqj .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#wmwobwlzqj .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#wmwobwlzqj .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#wmwobwlzqj .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#wmwobwlzqj .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#wmwobwlzqj .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#wmwobwlzqj .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#wmwobwlzqj .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#wmwobwlzqj .gt_from_md > :first-child {
  margin-top: 0;
}

#wmwobwlzqj .gt_from_md > :last-child {
  margin-bottom: 0;
}

#wmwobwlzqj .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#wmwobwlzqj .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#wmwobwlzqj .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#wmwobwlzqj .gt_row_group_first td {
  border-top-width: 2px;
}

#wmwobwlzqj .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#wmwobwlzqj .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#wmwobwlzqj .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#wmwobwlzqj .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#wmwobwlzqj .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#wmwobwlzqj .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#wmwobwlzqj .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#wmwobwlzqj .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#wmwobwlzqj .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#wmwobwlzqj .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#wmwobwlzqj .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#wmwobwlzqj .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#wmwobwlzqj .gt_left {
  text-align: left;
}

#wmwobwlzqj .gt_center {
  text-align: center;
}

#wmwobwlzqj .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#wmwobwlzqj .gt_font_normal {
  font-weight: normal;
}

#wmwobwlzqj .gt_font_bold {
  font-weight: bold;
}

#wmwobwlzqj .gt_font_italic {
  font-style: italic;
}

#wmwobwlzqj .gt_super {
  font-size: 65%;
}

#wmwobwlzqj .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#wmwobwlzqj .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#wmwobwlzqj .gt_indent_1 {
  text-indent: 5px;
}

#wmwobwlzqj .gt_indent_2 {
  text-indent: 10px;
}

#wmwobwlzqj .gt_indent_3 {
  text-indent: 15px;
}

#wmwobwlzqj .gt_indent_4 {
  text-indent: 20px;
}

#wmwobwlzqj .gt_indent_5 {
  text-indent: 25px;
}
</style>
<div class="inline-table"><table class="gt_table">
<thead class="gt_col_headings"><tr>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>OR</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>95% CI</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>p-value</strong></th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td class="gt_row gt_left">gender</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.88, 1.14</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">fever</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.86, 1.18</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">chills</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.89, 1.21</td>
<td class="gt_row gt_center">0.7</td>
</tr>
<tr>
<td class="gt_row gt_left">cough</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.96, 1.37</td>
<td class="gt_row gt_center">0.12</td>
</tr>
<tr>
<td class="gt_row gt_left">aches</td>
<td class="gt_row gt_center">0.93</td>
<td class="gt_row gt_center">0.76, 1.14</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left">vomit</td>
<td class="gt_row gt_center">1.09</td>
<td class="gt_row gt_center">0.96, 1.23</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left">age_cat</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">0-4</td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">5-9</td>
<td class="gt_row gt_center">0.94</td>
<td class="gt_row gt_center">0.77, 1.15</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">10-14</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.93, 1.41</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">15-19</td>
<td class="gt_row gt_center">0.99</td>
<td class="gt_row gt_center">0.79, 1.24</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">20-29</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.84, 1.26</td>
<td class="gt_row gt_center">0.8</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">30-49</td>
<td class="gt_row gt_center">1.06</td>
<td class="gt_row gt_center">0.85, 1.33</td>
<td class="gt_row gt_center">0.6</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">50-69</td>
<td class="gt_row gt_center">0.68</td>
<td class="gt_row gt_center">0.40, 1.13</td>
<td class="gt_row gt_center">0.14</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">70+</td>
<td class="gt_row gt_center">0.52</td>
<td class="gt_row gt_center">0.07, 3.19</td>
<td class="gt_row gt_center">0.5</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="4">
<sup class="gt_footnote_marks">1</sup> OR = Odds Ratio, CI = Confidence Interval</td>
    </tr></tfoot>
</table></div>
</div>
<p>También puedes combinar varias tablas de salida diferentes producidas por <strong>gtsummary</strong> con la función <code>tbl_merge()</code>. Ahora combinamos los resultados multivariables con los resultados <em>univariantes</em> de <strong>gtsummary</strong> que creamos <a href="univariate-and-multivariable-regression.html#reg_gt_uni">anteriormente</a>:</p>
<div class="sourceCode" id="cb1005"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## combine with univariate results </span>
<span class="fu">tbl_merge</span><span class="op">(</span>
  tbls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">univ_tab</span>, <span class="va">mv_tab</span><span class="op">)</span>,                          <span class="co"># combine</span>
  tab_spanner <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"**Univariate**"</span>, <span class="st">"**Multivariable**"</span><span class="op">)</span><span class="op">)</span> <span class="co"># set header names</span></code></pre></div>
<pre><code>## ! Use of `vars()` is now deprecated and support will soon be removed. Please replace calls to `vars()` with `c()`.
## ! Use of `vars()` is now deprecated and support will soon be removed. Please replace calls to `vars()` with `c()`.</code></pre>
<pre><code>## Warning: Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.
## Since gt v0.6.0 the `fmt_missing()` function is deprecated and will soon be
## removed.
## * Use the `sub_missing()` function instead.</code></pre>
<div id="ayixkobive" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#ayixkobive .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ayixkobive .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ayixkobive .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ayixkobive .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ayixkobive .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ayixkobive .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ayixkobive .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ayixkobive .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ayixkobive .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ayixkobive .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ayixkobive .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ayixkobive .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#ayixkobive .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ayixkobive .gt_from_md > :first-child {
  margin-top: 0;
}

#ayixkobive .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ayixkobive .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ayixkobive .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ayixkobive .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ayixkobive .gt_row_group_first td {
  border-top-width: 2px;
}

#ayixkobive .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ayixkobive .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ayixkobive .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ayixkobive .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ayixkobive .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ayixkobive .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ayixkobive .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ayixkobive .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ayixkobive .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ayixkobive .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ayixkobive .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ayixkobive .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ayixkobive .gt_left {
  text-align: left;
}

#ayixkobive .gt_center {
  text-align: center;
}

#ayixkobive .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ayixkobive .gt_font_normal {
  font-weight: normal;
}

#ayixkobive .gt_font_bold {
  font-weight: bold;
}

#ayixkobive .gt_font_italic {
  font-style: italic;
}

#ayixkobive .gt_super {
  font-size: 65%;
}

#ayixkobive .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#ayixkobive .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ayixkobive .gt_indent_1 {
  text-indent: 5px;
}

#ayixkobive .gt_indent_2 {
  text-indent: 10px;
}

#ayixkobive .gt_indent_3 {
  text-indent: 15px;
}

#ayixkobive .gt_indent_4 {
  text-indent: 20px;
}

#ayixkobive .gt_indent_5 {
  text-indent: 25px;
}
</style>
<div class="inline-table"><table class="gt_table">
<thead class="gt_col_headings">
<tr>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="2" colspan="1" scope="col"><strong>Characteristic</strong></th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="4" scope="colgroup">
        <span class="gt_column_spanner"><strong>Univariate</strong></span>
      </th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="3" scope="colgroup">
        <span class="gt_column_spanner"><strong>Multivariable</strong></span>
      </th>
    </tr>
<tr>
<th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>N</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>OR</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>95% CI</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>p-value</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>OR</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col">
<strong>95% CI</strong><sup class="gt_footnote_marks">1</sup>
</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>p-value</strong></th>
    </tr>
</thead>
<tbody class="gt_table_body">
<tr>
<td class="gt_row gt_left">gender</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.88, 1.13</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.88, 1.14</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">fever</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.85, 1.17</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">1.00</td>
<td class="gt_row gt_center">0.86, 1.18</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left">chills</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.89, 1.21</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.89, 1.21</td>
<td class="gt_row gt_center">0.7</td>
</tr>
<tr>
<td class="gt_row gt_left">cough</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.97, 1.37</td>
<td class="gt_row gt_center">0.11</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.96, 1.37</td>
<td class="gt_row gt_center">0.12</td>
</tr>
<tr>
<td class="gt_row gt_left">aches</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">0.93</td>
<td class="gt_row gt_center">0.76, 1.14</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">0.93</td>
<td class="gt_row gt_center">0.76, 1.14</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left">vomit</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center">1.09</td>
<td class="gt_row gt_center">0.96, 1.23</td>
<td class="gt_row gt_center">0.2</td>
<td class="gt_row gt_center">1.09</td>
<td class="gt_row gt_center">0.96, 1.23</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left">age_cat</td>
<td class="gt_row gt_center">4167</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">0-4</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center">—</td>
<td class="gt_row gt_center"></td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">5-9</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.94</td>
<td class="gt_row gt_center">0.77, 1.15</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">0.94</td>
<td class="gt_row gt_center">0.77, 1.15</td>
<td class="gt_row gt_center">0.5</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">10-14</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.93, 1.42</td>
<td class="gt_row gt_center">0.2</td>
<td class="gt_row gt_center">1.15</td>
<td class="gt_row gt_center">0.93, 1.41</td>
<td class="gt_row gt_center">0.2</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">15-19</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.99</td>
<td class="gt_row gt_center">0.80, 1.24</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">0.99</td>
<td class="gt_row gt_center">0.79, 1.24</td>
<td class="gt_row gt_center">&gt;0.9</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">20-29</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.84, 1.26</td>
<td class="gt_row gt_center">0.8</td>
<td class="gt_row gt_center">1.03</td>
<td class="gt_row gt_center">0.84, 1.26</td>
<td class="gt_row gt_center">0.8</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">30-49</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">1.07</td>
<td class="gt_row gt_center">0.85, 1.33</td>
<td class="gt_row gt_center">0.6</td>
<td class="gt_row gt_center">1.06</td>
<td class="gt_row gt_center">0.85, 1.33</td>
<td class="gt_row gt_center">0.6</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">50-69</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.68</td>
<td class="gt_row gt_center">0.41, 1.13</td>
<td class="gt_row gt_center">0.13</td>
<td class="gt_row gt_center">0.68</td>
<td class="gt_row gt_center">0.40, 1.13</td>
<td class="gt_row gt_center">0.14</td>
</tr>
<tr>
<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">70+</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center">0.53</td>
<td class="gt_row gt_center">0.07, 3.20</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">0.52</td>
<td class="gt_row gt_center">0.07, 3.19</td>
<td class="gt_row gt_center">0.5</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="8">
<sup class="gt_footnote_marks">1</sup> OR = Odds Ratio, CI = Confidence Interval</td>
    </tr></tfoot>
</table></div>
</div>
</div>
<div id="combinar-con-dplyr" class="section level4 unnumbered">
<h4>Combinar con <strong>dplyr</strong><a class="anchor" aria-label="anchor" href="#combinar-con-dplyr"><i class="fas fa-link"></i></a>
</h4>
<p>Una forma alternativa de combinar los resultados univariables y multivariables de <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>/<code>tidy()</code> es con las funciones join de <strong>dplyr</strong>.</p>
<ul>
<li>Unir los resultados univariantes de antes (<code>univ_tab_base</code>, que contiene los recuentos) con los resultados multivariables ordenados <code>mv_tab_base</code>
</li>
<li>Utiliza <code>select()</code> para mantener sólo las columnas que queremos, especificar tu orden y renombrarlas</li>
<li>Utiliza <code><a href="https://rdrr.io/r/base/Round.html">round()</a></code> con dos decimales en toda la columna que sea de tipo Double</li>
</ul>
<div class="sourceCode" id="cb1008"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## combine univariate and multivariable tables </span>
<span class="fu">left_join</span><span class="op">(</span><span class="va">univ_tab_base</span>, <span class="va">mv_tab_base</span>, by <span class="op">=</span> <span class="st">"term"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co">## choose columns and rename them</span>
  <span class="fu">select</span><span class="op">(</span> <span class="co"># new name =  old name</span>
    <span class="st">"characteristic"</span> <span class="op">=</span> <span class="va">term</span>, 
    <span class="st">"recovered"</span>      <span class="op">=</span> <span class="st">"0"</span>, 
    <span class="st">"dead"</span>           <span class="op">=</span> <span class="st">"1"</span>, 
    <span class="st">"univ_or"</span>        <span class="op">=</span> <span class="va">estimate.x</span>, 
    <span class="st">"univ_ci_low"</span>    <span class="op">=</span> <span class="va">conf.low.x</span>, 
    <span class="st">"univ_ci_high"</span>   <span class="op">=</span> <span class="va">conf.high.x</span>,
    <span class="st">"univ_pval"</span>      <span class="op">=</span> <span class="va">p.value.x</span>, 
    <span class="st">"mv_or"</span>          <span class="op">=</span> <span class="va">estimate.y</span>, 
    <span class="st">"mvv_ci_low"</span>     <span class="op">=</span> <span class="va">conf.low.y</span>, 
    <span class="st">"mv_ci_high"</span>     <span class="op">=</span> <span class="va">conf.high.y</span>,
    <span class="st">"mv_pval"</span>        <span class="op">=</span> <span class="va">p.value.y</span> 
  <span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">is.double</span><span class="op">)</span>, <span class="va">round</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>   </code></pre></div>
<pre><code>## # A tibble: 20 x 11
##    characteristic recov~1  dead univ_or univ_~2 univ_~3 univ_~4 mv_or mvv_c~5
##    &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;
##  1 (Intercept)        909  1168    1.28    1.18    1.4     0     1.07    0.83
##  2 gender             916  1174    1       0.88    1.13    0.97  1       0.88
##  3 (Intercept)        340   436    1.28    1.11    1.48    0     1.07    0.83
##  4 fever             1485  1906    1       0.85    1.17    0.99  1       0.86
##  5 (Intercept)       1472  1877    1.28    1.19    1.37    0     1.07    0.83
##  6 chills             353   465    1.03    0.89    1.21    0.68  1.03    0.89
##  7 (Intercept)        272   309    1.14    0.97    1.34    0.13  1.07    0.83
##  8 cough             1553  2033    1.15    0.97    1.37    0.11  1.15    0.96
##  9 (Intercept)       1636  2114    1.29    1.21    1.38    0     1.07    0.83
## 10 aches              189   228    0.93    0.76    1.14    0.51  0.93    0.76
## 11 (Intercept)        931  1144    1.23    1.13    1.34    0     1.07    0.83
## 12 vomit              894  1198    1.09    0.96    1.23    0.17  1.09    0.96
## 13 (Intercept)        338   427    1.26    1.1     1.46    0     1.07    0.83
## 14 age_cat5-9         365   433    0.94    0.77    1.15    0.54  0.94    0.77
## 15 age_cat10-14       273   396    1.15    0.93    1.42    0.2   1.15    0.93
## 16 age_cat15-19       238   299    0.99    0.8     1.24    0.96  0.99    0.79
## 17 age_cat20-29       345   448    1.03    0.84    1.26    0.79  1.03    0.84
## 18 age_cat30-49       228   307    1.07    0.85    1.33    0.58  1.06    0.85
## 19 age_cat50-69        35    30    0.68    0.41    1.13    0.13  0.68    0.4 
## 20 age_cat70+           3     2    0.53    0.07    3.2     0.49  0.52    0.07
## # ... with 2 more variables: mv_ci_high &lt;dbl&gt;, mv_pval &lt;dbl&gt;, and
## #   abbreviated variable names 1: recovered, 2: univ_ci_low,
## #   3: univ_ci_high, 4: univ_pval, 5: mvv_ci_low</code></pre>
<!-- ======================================================= -->
</div>
</div>
</div>
<div id="forest-plot" class="section level2" number="18.5">
<h2>
<span class="header-section-number">18.5</span> Forest plot<a class="anchor" aria-label="anchor" href="#forest-plot"><i class="fas fa-link"></i></a>
</h2>
<p>Esta sección muestra cómo producir un gráfico con los resultados de tu regresión.
Hay dos opciones, puedes construir un gráfico tú mismo usando <strong>ggplot2</strong> o usar un metapaquete llamado <strong>easystats</strong> (un paquete que incluye muchos paquetes).</p>
<p>Consulta la página sobre <a href="ggplot-basics.html#ggplot-basics">Conceptos básicos de ggplot</a> si no estás familiarizado con el paquete de gráficos <strong>ggplot2</strong>.</p>
<!-- ======================================================= -->
<div id="paquete-ggplot2" class="section level3 unnumbered">
<h3>Paquete <strong>ggplot2</strong><a class="anchor" aria-label="anchor" href="#paquete-ggplot2"><i class="fas fa-link"></i></a>
</h3>
<p>Puedes construir un gráfico de bosque con <code>ggplot()</code> trazando elementos de los resultados de la regresión multivariable. Añada las capas de los gráficos utilizando estos “geoms”:</p>
<ul>
<li>estimaciones con <code>geom_point()</code>
</li>
<li>intervalos de confianza con <code>geom_errorbar()</code>
</li>
<li>una línea vertical en OR = 1 con <code>geom_vline()</code>
</li>
</ul>
<p>Antes de trazar, es posible que desee utilizar <code>fct_relevel()</code> del paquete <strong>forcats</strong> para establecer el orden de las variables/niveles en el eje y. <code>ggplot()</code> puede mostrarlos en orden alfanumérico, lo que no funcionaría bien para estos valores de categoría de edad (“30” aparecería antes de “5”). Mira la página sobre <a href="#factors">Factores</a> para más detalles.</p>
<div class="sourceCode" id="cb1010"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## remove the intercept term from your multivariable results</span>
<span class="va">mv_tab_base</span> <span class="op">%&gt;%</span> 
  
  <span class="co">#set order of levels to appear along y-axis</span>
  <span class="fu">mutate</span><span class="op">(</span>term <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span>
    <span class="va">term</span>,
    <span class="st">"vomit"</span>, <span class="st">"gender"</span>, <span class="st">"fever"</span>, <span class="st">"cough"</span>, <span class="st">"chills"</span>, <span class="st">"aches"</span>,
    <span class="st">"age_cat5-9"</span>, <span class="st">"age_cat10-14"</span>, <span class="st">"age_cat15-19"</span>, <span class="st">"age_cat20-29"</span>,
    <span class="st">"age_cat30-49"</span>, <span class="st">"age_cat50-69"</span>, <span class="st">"age_cat70+"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="co"># remove "intercept" row from plot</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">term</span> <span class="op">!=</span> <span class="st">"(Intercept)"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  
  <span class="co">## plot with variable on the y axis and estimate (OR) on the x axis</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">estimate</span>, y <span class="op">=</span> <span class="va">term</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  
  <span class="co">## show the estimate as a point</span>
  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  
  <span class="co">## add in an error bar for the confidence intervals</span>
  <span class="fu">geom_errorbar</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">conf.low</span>, xmax <span class="op">=</span> <span class="va">conf.high</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  
  <span class="co">## show where OR = 1 is for reference as a dashed line</span>
  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">1</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/ggplot_forest-1.png" width="672"></div>
<!-- ======================================================= -->
</div>
<div id="paquetes-easystats" class="section level3 unnumbered">
<h3>paquetes <strong>easystats</strong><a class="anchor" aria-label="anchor" href="#paquetes-easystats"><i class="fas fa-link"></i></a>
</h3>
<p>Una alternativa, si no deseas el nivel de control fino que proporciona <strong>ggplot2</strong>, es utilizar una combinación de paquetes <strong>easystats</strong>.</p>
<p>La función <code>model_parameters()</code> del paquete <strong>parameters</strong> hace el equivalente de la función <code>tidy()</code> del paquete <strong>broom</strong>. El paquete <strong>see</strong> acepta esos resultados y crea por defecto un forest plot como un objeto <code>ggplot()</code>.</p>
<div class="sourceCode" id="cb1011"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">easystats</span><span class="op">)</span>

<span class="co">## remove the intercept term from your multivariable results</span>
<span class="va">final_mv_reg</span> <span class="op">%&gt;%</span> 
  <span class="fu">model_parameters</span><span class="op">(</span>exponentiate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/easystats_forest-1.png" width="672"></div>
<!-- ======================================================= -->
</div>
</div>
<div id="resources-11" class="section level2" number="18.6">
<h2>
<span class="header-section-number">18.6</span> Recursos<a class="anchor" aria-label="anchor" href="#resources-11"><i class="fas fa-link"></i></a>
</h2>
<p>El contenido de esta página se ha basado en estos recursos y viñetas en línea:</p>
<p><a href="https://www.datacamp.com/community/tutorials/linear-regression-R">Regresión lineal en R</a></p>
<p><a href="http://www.danieldsjoberg.com/gtsummary/articles/tbl_regression.html">gtsummary</a></p>
<p><a href="https://stats.idre.ucla.edu/other/dae/">Página de estadísticas de la UCLA</a></p>
<p><a href="http://www.sthda.com/english/articles/36-classification-methods-essentials/150-stepwise-logistic-regression-essentials-in-r/">regresión escalonada sthda</a></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="simple-statistical-tests.html"><span class="header-section-number">17</span> Tests estadísticos sencillos</a></div>
<div class="next"><a href="missing-data.html"><span class="header-section-number">19</span> Borrador. Valores faltantes</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#univariate-and-multivariable-regression"><span class="header-section-number">18</span> Borrador. Regresión univariante y multivariable</a></li>
<li>
<a class="nav-link" href="#preparation-9"><span class="header-section-number">18.1</span> Preparación</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#cargar-paquetes-11">Cargar paquetes</a></li>
<li><a class="nav-link" href="#importar-datos-10">Importar datos</a></li>
<li><a class="nav-link" href="#datos-limpios">Datos limpios</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#univariate"><span class="header-section-number">18.2</span> Univariante</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#r-base-1">R base</a></li>
<li><a class="nav-link" href="#reg_gt_uni">paquete gtsummary</a></li>
</ul>
</li>
<li><a class="nav-link" href="#stratified"><span class="header-section-number">18.3</span> Estratificado</a></li>
<li>
<a class="nav-link" href="#multivariable"><span class="header-section-number">18.4</span> Multivariable</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#realizar-el-multivariable">Realizar el multivariable</a></li>
<li><a class="nav-link" href="#combine-univariate-and-multivariable">Combine univariate and multivariable</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#forest-plot"><span class="header-section-number">18.5</span> Forest plot</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#paquete-ggplot2">Paquete ggplot2</a></li>
<li><a class="nav-link" href="#paquetes-easystats">paquetes easystats</a></li>
</ul>
</li>
<li><a class="nav-link" href="#resources-11"><span class="header-section-number">18.6</span> Recursos</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>El manual de R para Epi’s</strong>" was written by El equipo del manual. It was last built on 2022-10-18.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
